<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Add EMBOSS-inspired theoretical physicochemical properties
using the Peptides library to an AnnotatedDataFrame — add_physicochemical_properties_to_HMMER_tbl • HMMERutils</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Add EMBOSS-inspired theoretical physicochemical properties
using the Peptides library to an AnnotatedDataFrame — add_physicochemical_properties_to_HMMER_tbl"><meta property="og:description" content="Add EMBOSS-inspired theoretical physicochemical properties
using the Peptides library to an AnnotatedDataFrame"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">HMMERutils</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/HMMERutils.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/currocam/HMMERutils/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Add EMBOSS-inspired theoretical physicochemical properties
using the <code>Peptides</code> library to an <code>AnnotatedDataFrame</code></h1>
    <small class="dont-index">Source: <a href="https://github.com/currocam/HMMERutils/blob/HEAD/R/add_physicochemical_properties_to_HMMER_tbl.R" class="external-link"><code>R/add_physicochemical_properties_to_HMMER_tbl.R</code></a></small>
    <div class="hidden name"><code>add_physicochemical_properties_to_HMMER_tbl.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Add EMBOSS-inspired theoretical physicochemical properties
using the <code>Peptides</code> library to an <code>AnnotatedDataFrame</code></p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">add_physicochemical_properties_to_HMMER_tbl</span><span class="op">(</span><span class="va">data</span>, colname <span class="op">=</span> <span class="st">"hits.fullfasta"</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>data</dt>
<dd><p>A Data Frame containig the indicated
column with protein sequences.</p></dd>


<dt>colname</dt>
<dd><p>A single character vector with the
name of the column with the protein sequences.</p>
<p>This documentation is based on the following documentation:
https://github.com/dosorio/Peptides/</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A Data Frame with new columns with
the theoretical physicochemical properties</p>
    </div>
    <div id="theoretical-properties">
    <h2>Theoretical properties</h2>
    

<ul><li><p>molecular.weight: sum of the masses of each atom constituting a
molecule (Da) using the same formulas and weights as ExPASy's.</p></li>
<li><p>charge: net charge of a protein sequence based on the
Henderson-Hasselbalch equation using Lehninger pKa scale.</p></li>
<li><p>pI:  isoelectric point calculated as in EMBOSS PEPSTATS.</p></li>
<li><p>mz: mass over charge ratio (m/z) for peptides,
as measured in mass spectrometry.</p></li>
<li><p>aIndex: aliphatic index of a protein. The aindex is defined as the
relative volume occupied by aliphatic side chains (Alanine, Valine,
Isoleucine, and Leucine). It may be regarded as a positive factor
for the increase of thermostability of globular proteins.</p></li>
<li><p>boman: potential protein interaction index . The index is equal to the
sum of the solubility values for all residues in a sequence, it might give
an overall estimate of the potential of a peptide to bind to membranes or
other proteins as receptors, to normalize it is divided by the number of
residues. A protein have high binding potential if the index value is
higher than 2.48.</p></li>
<li><p>hydrophobicity: GRAVY hydrophobicity index of an amino acids sequence using
KyteDoolittle hydophobicity scale.</p></li>
<li><p>instaIndex: Guruprasad's instability index.
This index predicts the stability of a protein based
on its amino acid composition, a protein whose instability
index is smaller than 40 is predicted as stable, a value
above 40 predicts that the protein may be unstable.</p></li>
<li><p>STYNQW: Percentage of amino acids (S + T + Y + N + Q + W)</p></li>
<li><p>Tiny: Percentage of amino acids (A + C + G + S + T)</p></li>
<li><p>Small: Percentage of amino acids (A + B + C + D + G + N + P + S + T + V)</p></li>
<li><p>Aliphatic: Percentage of amino acids (A + I + L + V)</p></li>
<li><p>Aromatic: Percentage of amino acids (F + H + W + Y)</p></li>
<li><p>Nonpolar`: Percentage of amino acids
(A + C + F + G + I + L + M + P + V + W + Y)</p></li>
<li><p>Polar: Percentage of amino acids (D + E + H + K + N + Q + R + S + T + Z)</p></li>
<li><p>Charged: Percentage of amino acids (B + D + E + H + K + R + Z)</p></li>
<li><p>Basic: Percentage of amino acids (H + K + R)</p></li>
<li><p>Acidic: Percentage of amino acids (B + D + E + Z)</p></li>
</ul></div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Francisco de Borja Campuzano Jiménez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

