<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to HMMERutils • HMMERutils</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to HMMERutils">
<meta property="og:description" content="HMMERutils">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">HMMERutils</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/HMMERutils.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/currocam/HMMERutils/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to HMMERutils</h1>
                        <h4 data-toc-skip class="author">Curro Campuzano Jiménez</h4>
                  <a class="author_email" href="mailto:#"></a><a href="mailto:campuzanocurro@gmail.com" class="email">campuzanocurro@gmail.com</a>
      
                  
            <h4 data-toc-skip class="date">18 November 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/currocam/HMMERutils/blob/HEAD/vignettes/HMMERutils.Rmd" class="external-link"><code>vignettes/HMMERutils.Rmd</code></a></small>
      <div class="hidden name"><code>HMMERutils.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="basics">Basics<a class="anchor" aria-label="anchor" href="#basics"></a>
</h2>
<div class="section level3">
<h3 id="install-hmmerutils">Install <code>HMMERutils</code><a class="anchor" aria-label="anchor" href="#install-hmmerutils"></a>
</h3>
<p><code>R</code> is an open-source statistical environment that can be easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.16/HMMERutils" class="external-link">HMMERutils</a></em> is an <code>R</code> package available via the <a href="http://bioconductor.org" class="external-link">Bioconductor</a> repository for packages. <code>R</code> can be installed on any operating system from <a href="https://cran.r-project.org/" class="external-link">CRAN</a> after which you can install <em><a href="https://bioconductor.org/packages/3.16/HMMERutils" class="external-link">HMMERutils</a></em> by using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"HMMERutils"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check that you have a valid Bioconductor installation</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="required-knowledge">Required knowledge<a class="anchor" aria-label="anchor" href="#required-knowledge"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.16/HMMERutils" class="external-link">HMMERutils</a></em> is based on many other packages and in particular those that have implemented the infrastructure needed for dealing with protein sequences data. That is, packages like <em><a href="https://bioconductor.org/packages/3.16/Biostrings" class="external-link">Biostrings</a></em> or <em><a href="https://bioconductor.org/packages/3.16/Peptides" class="external-link">Peptides</a></em>.</p>
<p>If you are asking yourself the question “Where do I start using Bioconductor?” you might be interested in <a href="http://lcolladotor.github.io/2014/10/16/startbioc/#.VkOKbq6rRuU" class="external-link">this post</a>.</p>
</div>
<div class="section level3">
<h3 id="asking-for-help">Asking for help<a class="anchor" aria-label="anchor" href="#asking-for-help"></a>
</h3>
<p>As package developers, we try to explain clearly how to use our packages and in which order to use the functions. But <code>R</code> and <code>Bioconductor</code> have a steep learning curve so it is critical to learn where to ask for help. The blog post quoted above mentions some but we would like to highlight the <a href="https://support.bioconductor.org/" class="external-link">Bioconductor support site</a> as the main resource for getting help: remember to use the <code>HMMERutils</code> tag and check <a href="https://support.bioconductor.org/t/HMMERutils/" class="external-link">the older posts</a>. Other alternatives are available such as creating GitHub issues and tweeting. However, please note that if you want to receive the help you should adhere to the <a href="http://www.bioconductor.org/help/support/posting-guide/" class="external-link">posting guidelines</a>. It is particularly critical that you provide a small reproducible example and your session information so package developers can track down the source of the error.</p>
</div>
<div class="section level3">
<h3 id="citing-hmmerutils">Citing <code>HMMERutils</code><a class="anchor" aria-label="anchor" href="#citing-hmmerutils"></a>
</h3>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.16/HMMERutils" class="external-link">HMMERutils</a></em> will be useful for your research. Please use the following information to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Citation info</span></span>
<span><span class="co"># citation("HMMERutils")</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="quick-start-using-hmmerutils">Quick start using <code>HMMERutils</code><a class="anchor" aria-label="anchor" href="#quick-start-using-hmmerutils"></a>
</h2>
<p>The objective of this article is to show the main functionalities of the utilsHMMER library using a case study. We will search for homologous sequences for the whole SARS-CoV-2 reference proteome.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/currocam/HMMERutils" class="external-link">"HMMERutils"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/Biostrings" class="external-link">"Biostrings"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org" class="external-link">"dplyr"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="download-the-sars-cov-2-reference-proteome">Download the SARS-CoV-2 reference proteome<a class="anchor" aria-label="anchor" href="#download-the-sars-cov-2-reference-proteome"></a>
</h4>
<p>First, let’s read the reference proteome for the Sars_CoV-2. We will read the compressed file directly from UniProt with the <code>readAAStringSet</code> function from the <em><a href="https://bioconductor.org/packages/3.16/Biostrings" class="external-link">Biostrings</a></em> library.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">uniprot.url</span> <span class="op">&lt;-</span> <span class="st">"https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/pan_proteomes/UP000464024.fasta.gz"</span></span>
<span></span>
<span><span class="va">sars.cov.fasta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">readAAStringSet</a></span><span class="op">(</span><span class="va">uniprot.url</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="preprocess-sequences-and-select-case-studies">Preprocess sequences and select case studies<a class="anchor" aria-label="anchor" href="#preprocess-sequences-and-select-case-studies"></a>
</h4>
<p>Now, we have an <code>AAStringSet</code> stored in the variable <code>sars.cov.fasta</code> and we can access the sequences it contains using the <code>$</code> operator and the sequence header. However, headers usually contain, as in this case, a lot of information that is not useful right now such as descriptions.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sars.cov.fasta</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "tr|A0A166ZLE2|A0A166ZLE2_9NIDO Accessory protein 7b OS=Bat coronavirus OX=1508220 GN=ORF7b PE=4 SV=1 UPId=UP000102490 PPId=UP000464024"                             </span></span>
<span><span class="co">#&gt;  [2] "tr|A0A166ZL78|A0A166ZL78_9NIDO 3b protein OS=Bat coronavirus OX=1508220 GN=ORF3b PE=4 SV=1 UPId=UP000102490 PPId=UP000464024"                                       </span></span>
<span><span class="co">#&gt;  [3] "tr|A0A0U1WHH4|A0A0U1WHH4_SARS Accessory protein 7b OS=BtRf-BetaCoV/JL2012 OX=1503299 PE=4 SV=1 UPId=UP000123530 PPId=UP000464024"                                   </span></span>
<span><span class="co">#&gt;  [4] "tr|A0A0U1WHH1|A0A0U1WHH1_SARS ORF3b OS=BtRf-BetaCoV/JL2012 OX=1503299 PE=4 SV=1 UPId=UP000123530 PPId=UP000464024"                                                  </span></span>
<span><span class="co">#&gt;  [5] "sp|P0DTC3|AP3A_SARS2 ORF3a protein OS=Severe acute respiratory syndrome coronavirus 2 OX=2697049 GN=3a PE=1 SV=1 UPId=UP000464024 PPId=UP000464024"                 </span></span>
<span><span class="co">#&gt;  [6] "sp|P0DTC9|NCAP_SARS2 Nucleoprotein OS=Severe acute respiratory syndrome coronavirus 2 OX=2697049 GN=N PE=1 SV=1 UPId=UP000464024 PPId=UP000464024"                  </span></span>
<span><span class="co">#&gt;  [7] "sp|P0DTC6|NS6_SARS2 ORF6 protein OS=Severe acute respiratory syndrome coronavirus 2 OX=2697049 GN=6 PE=1 SV=1 UPId=UP000464024 PPId=UP000464024"                    </span></span>
<span><span class="co">#&gt;  [8] "sp|P0DTC7|NS7A_SARS2 ORF7a protein OS=Severe acute respiratory syndrome coronavirus 2 OX=2697049 GN=7a PE=1 SV=1 UPId=UP000464024 PPId=UP000464024"                 </span></span>
<span><span class="co">#&gt;  [9] "sp|P0DTD8|NS7B_SARS2 ORF7b protein OS=Severe acute respiratory syndrome coronavirus 2 OX=2697049 GN=7b PE=1 SV=1 UPId=UP000464024 PPId=UP000464024"                 </span></span>
<span><span class="co">#&gt; [10] "sp|P0DTC8|NS8_SARS2 ORF8 protein OS=Severe acute respiratory syndrome coronavirus 2 OX=2697049 GN=8 PE=1 SV=1 UPId=UP000464024 PPId=UP000464024"                    </span></span>
<span><span class="co">#&gt; [11] "sp|A0A663DJA2|ORF10_SARS2 Putative ORF10 protein OS=Severe acute respiratory syndrome coronavirus 2 OX=2697049 GN=ORF10 PE=5 SV=1 UPId=UP000464024 PPId=UP000464024"</span></span>
<span><span class="co">#&gt; [12] "sp|P0DTF1|ORF3B_SARS2 Putative ORF3b protein OS=Severe acute respiratory syndrome coronavirus 2 OX=2697049 PE=5 SV=1 UPId=UP000464024 PPId=UP000464024"             </span></span>
<span><span class="co">#&gt; [13] "sp|P0DTG1|ORF3C_SARS2 ORF3c protein OS=Severe acute respiratory syndrome coronavirus 2 OX=2697049 PE=4 SV=1 UPId=UP000464024 PPId=UP000464024"                      </span></span>
<span><span class="co">#&gt; [14] "sp|P0DTG0|ORF3D_SARS2 Putative ORF3d protein OS=Severe acute respiratory syndrome coronavirus 2 OX=2697049 PE=5 SV=1 UPId=UP000464024 PPId=UP000464024"             </span></span>
<span><span class="co">#&gt; [15] "sp|P0DTD2|ORF9B_SARS2 ORF9b protein OS=Severe acute respiratory syndrome coronavirus 2 OX=2697049 GN=9b PE=1 SV=1 UPId=UP000464024 PPId=UP000464024"                </span></span>
<span><span class="co">#&gt; [16] "sp|P0DTD3|ORF9C_SARS2 Putative ORF9c protein OS=Severe acute respiratory syndrome coronavirus 2 OX=2697049 GN=9c PE=5 SV=1 UPId=UP000464024 PPId=UP000464024"       </span></span>
<span><span class="co">#&gt; [17] "sp|P0DTD1|R1AB_SARS2 Replicase polyprotein 1ab OS=Severe acute respiratory syndrome coronavirus 2 OX=2697049 GN=rep PE=1 SV=1 UPId=UP000464024 PPId=UP000464024"    </span></span>
<span><span class="co">#&gt; [18] "sp|P0DTC1|R1A_SARS2 Replicase polyprotein 1a OS=Severe acute respiratory syndrome coronavirus 2 OX=2697049 PE=1 SV=1 UPId=UP000464024 PPId=UP000464024"             </span></span>
<span><span class="co">#&gt; [19] "sp|P0DTC2|SPIKE_SARS2 Spike glycoprotein OS=Severe acute respiratory syndrome coronavirus 2 OX=2697049 GN=S PE=1 SV=1 UPId=UP000464024 PPId=UP000464024"            </span></span>
<span><span class="co">#&gt; [20] "sp|P0DTC4|VEMP_SARS2 Envelope small membrane protein OS=Severe acute respiratory syndrome coronavirus 2 OX=2697049 GN=E PE=1 SV=1 UPId=UP000464024 PPId=UP000464024"</span></span>
<span><span class="co">#&gt; [21] "sp|P0DTC5|VME1_SARS2 Membrane protein OS=Severe acute respiratory syndrome coronavirus 2 OX=2697049 GN=M PE=1 SV=1 UPId=UP000464024 PPId=UP000464024"</span></span></code></pre></div>
<p>HMMERutils has a very convenient function for this purpose which is <code><a href="../reference/parse_fasta_headers.html">parse_fasta_headers()</a></code>. Note that we use the <code>name</code> function to access the sequence names and the <code>%&gt;%</code> operator to pass the result of one function/argument to the other in the sequence</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sars.cov.fasta</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sars.cov.fasta</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="../reference/parse_fasta_headers.html">parse_fasta_headers</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "A0A166ZLE2_9NIDO" "A0A166ZL78_9NIDO" "A0A0U1WHH4_SARS"  "A0A0U1WHH1_SARS" </span></span>
<span><span class="co">#&gt;  [5] "AP3A_SARS2"       "NCAP_SARS2"       "NS6_SARS2"        "NS7A_SARS2"      </span></span>
<span><span class="co">#&gt;  [9] "NS7B_SARS2"       "NS8_SARS2"        "ORF10_SARS2"      "ORF3B_SARS2"     </span></span>
<span><span class="co">#&gt; [13] "ORF3C_SARS2"      "ORF3D_SARS2"      "ORF9B_SARS2"      "ORF9C_SARS2"     </span></span>
<span><span class="co">#&gt; [17] "R1AB_SARS2"       "R1A_SARS2"        "SPIKE_SARS2"      "VEMP_SARS2"      </span></span>
<span><span class="co">#&gt; [21] "VME1_SARS2"</span></span></code></pre></div>
<p>We can take a look at the contents of the fasta file by “showing” it (calling the <code>show</code> is done implicitly):</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sars.cov.fasta</span></span>
<span><span class="co">#&gt; AAStringSet object of length 21:</span></span>
<span><span class="co">#&gt;      width seq                                              names               </span></span>
<span><span class="co">#&gt;  [1]    52 MNELTLIDFYLCFLAFLLFLVLI...SLELQDIEEPCNKVFETYRSVL A0A166ZLE2_9NIDO</span></span>
<span><span class="co">#&gt;  [2]   114 MMPTTLFAGIHIIMTTVYHTTVS...LTSLLKIRRMCKYTQSTALQEL A0A166ZL78_9NIDO</span></span>
<span><span class="co">#&gt;  [3]    48 MNELTLIDFYLCFLAFLLFLVLIMLIIFWFSLELQDIEEPCNKVFETL A0A0U1WHH4_SARS</span></span>
<span><span class="co">#&gt;  [4]   114 MMPTTLFAGIHIIMTTVYHTTVS...LTSLLKIRRMCKYTQSTALQEL A0A0U1WHH1_SARS</span></span>
<span><span class="co">#&gt;  [5]   275 MDLFMRIFTIGTVTLKQGEIKDA...GVVNPVMEPIYDEPTTTTSVPL AP3A_SARS2</span></span>
<span><span class="co">#&gt;  ...   ... ...</span></span>
<span><span class="co">#&gt; [17]  7096 MESLVPGFNEKTHVQLSLPVLQV...KGRLIIRENNRVVISSDVLVNN R1AB_SARS2</span></span>
<span><span class="co">#&gt; [18]  4405 MESLVPGFNEKTHVQLSLPVLQV...DQLREPMLQSADAQSFLNGFAV R1A_SARS2</span></span>
<span><span class="co">#&gt; [19]  1273 MFVFLVLLPLVSSQCVNLTTRTQ...SCCKFDEDDSEPVLKGVKLHYT SPIKE_SARS2</span></span>
<span><span class="co">#&gt; [20]    75 MYSFVSEETGTLIVNSVLLFLAF...PSFYVYSRVKNLNSSRVPDLLV VEMP_SARS2</span></span>
<span><span class="co">#&gt; [21]   222 MADSNGTITVEELKKLLEQWNLV...IGNYKLNTDHSSSSDNIALLVQ VME1_SARS2</span></span></code></pre></div>
<p>Now, we are going to select a series of proteins that are of interest to us for this case study:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">VEMP_SARS</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">sars.cov.fasta</span><span class="op">$</span><span class="va">VEMP_SARS2</span><span class="op">)</span> <span class="co"># Envelope small membrane protein</span></span>
<span><span class="va">NS8A_SARS</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">sars.cov.fasta</span><span class="op">$</span><span class="va">NS8_SARS2</span><span class="op">)</span> <span class="co"># ORF8 protein</span></span>
<span><span class="va">NS7B_SARS</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">sars.cov.fasta</span><span class="op">$</span><span class="va">NS7B_SARS</span><span class="op">)</span> <span class="co"># Protein non-structural 7b</span></span>
<span><span class="va">R1AB_SARS2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">sars.cov.fasta</span><span class="op">$</span><span class="va">R1AB_SARS2</span><span class="op">)</span> <span class="co"># Replicase polyprotein 1ab OS=Severe acute respirator</span></span></code></pre></div>
<p>All these proteins have in common being small in size (it will speed up some computational times) and represent different possibilities: we have from membrane proteins (which we would expect to show homology with other membrane proteins), proteins similar to each other (NS8A_SARS and NS8B_SARS), non-structural proteins, and uncharacterized proteins.</p>
</div>
<div class="section level4">
<h4 id="proteins-homologous-to-vemp_sars-using-phmmer">Proteins homologous to VEMP_SARS using PHMMER<a class="anchor" aria-label="anchor" href="#proteins-homologous-to-vemp_sars-using-phmmer"></a>
</h4>
<p>The most frequently used algorithm of HMMER is phmmer. This algorithm allows us to search for homologous sequences using a protein sequence as a target against protein sequence databases.</p>
<p>We can post a query to the HMMER web server and read the data into a user-friendly DataFrame by using the “search_” family of functions that HMMERutils provides.</p>
<p>Let’s start by searching for proteins homologous to VEMP_SARS (a envelope small membrane protein) against the Swissprot database</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hmmer.VEMP_SARS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/search_phmmer.html">search_phmmer</a></span><span class="op">(</span></span>
<span>    seq <span class="op">=</span> <span class="va">VEMP_SARS</span>, seqdb <span class="op">=</span> <span class="st">"swissprot"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Let’s inspect the content of the new variable we have created. We have two types of variables: the first 21 columns give us information about the query <em>itself</em> (such as the algorithm or the unique identifier), while the rest give us information about the <em>content</em> of the query.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="va">hmmer.VEMP_SARS</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">21</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 43</span></span>
<span><span class="co">#&gt; Columns: 21</span></span>
<span><span class="co">#&gt; $ algorithm         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "phmmer", "phmmer", "phmmer", "phmmer", "phmmer", "p…</span></span>
<span><span class="co">#&gt; $ uuid              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "31D3D19C-6745-11ED-9EE3-3AE29E870EE0", "31D3D19C-67…</span></span>
<span><span class="co">#&gt; $ stats.page        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1…</span></span>
<span><span class="co">#&gt; $ stats.nhits       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 43, 43, 43, 43, 43, 43, 43, 43, 43, 43, 43, 43, 43, …</span></span>
<span><span class="co">#&gt; $ stats.elapsed     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "0.05", "0.05", "0.05", "0.05", "0.05", "0.05", "0.0…</span></span>
<span><span class="co">#&gt; $ stats.Z           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 565928, 565928, 565928, 565928, 565928, 565928, 5659…</span></span>
<span><span class="co">#&gt; $ stats.Z_setby     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0…</span></span>
<span><span class="co">#&gt; $ stats.n_past_msv  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 24059, 24059, 24059, 24059, 24059, 24059, 24059, 240…</span></span>
<span><span class="co">#&gt; $ stats.unpacked    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 43, 43, 43, 43, 43, 43, 43, 43, 43, 43, 43, 43, 43, …</span></span>
<span><span class="co">#&gt; $ stats.nseqs       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 565928, 565928, 565928, 565928, 565928, 565928, 5659…</span></span>
<span><span class="co">#&gt; $ stats.user        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0…</span></span>
<span><span class="co">#&gt; $ stats.domZ_setby  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0…</span></span>
<span><span class="co">#&gt; $ stats.n_past_bias <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 14018, 14018, 14018, 14018, 14018, 14018, 14018, 140…</span></span>
<span><span class="co">#&gt; $ stats.sys         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0…</span></span>
<span><span class="co">#&gt; $ stats.n_past_fwd  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60, …</span></span>
<span><span class="co">#&gt; $ stats.total       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1…</span></span>
<span><span class="co">#&gt; $ stats.nmodels     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1…</span></span>
<span><span class="co">#&gt; $ stats.nincluded   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 35, 35, 35, 35, 35, 35, 35, 35, 35, 35, 35, 35, 35, …</span></span>
<span><span class="co">#&gt; $ stats.n_past_vit  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 902, 902, 902, 902, 902, 902, 902, 902, 902, 902, 90…</span></span>
<span><span class="co">#&gt; $ stats.nreported   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 43, 43, 43, 43, 43, 43, 43, 43, 43, 43, 43, 43, 43, …</span></span>
<span><span class="co">#&gt; $ stats.domZ        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 43, 43, 43, 43, 43, 43, 43, 43, 43, 43, 43, 43, 43, …</span></span></code></pre></div>
<p>As you can see, all rows have the same values for all the previous variables. That is a deliberate decision: each row represents one sequence hit that we get by doing just <strong>one</strong> query.</p>
<p>Most of the parameters contained in <code>stats</code> are for internal use and we are only interested in a small part of them. For example, by studying the table above, we can know that HMMER found 43 hits (<code>nhits</code>) out of a total of 565928 sequences available (<code>Z</code>) in the SwissProt and that a single HMM, Hidden Markov Model, (<code>nmodels</code>) has been used in the search (built from our input sequence since we are using PHMMER algorithm). In case you are interested in learning more about how the results data are structured, we invite you to read the appendix of the <a href="https://hmmer-web-docs.readthedocs.io/en/latest/appendices.html" class="external-link">HMMER web server</a>.</p>
<p>Let’s explore the rest of the Data Frame:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="va">hmmer.VEMP_SARS</span><span class="op">[</span>, <span class="fl">22</span><span class="op">:</span><span class="fl">47</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 43</span></span>
<span><span class="co">#&gt; Columns: 26</span></span>
<span><span class="co">#&gt; $ hits.archScore <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "1", "1", "1", "1", "1", "2", "2", "2", "0", "0", "0", …</span></span>
<span><span class="co">#&gt; $ hits.ph        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Pisuviricota", "Pisuviricota", "Pisuviricota", "Pisuvi…</span></span>
<span><span class="co">#&gt; $ hits.arch      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "PF02723.17", "PF02723.17", "PF02723.17", "PF02723.17",…</span></span>
<span><span class="co">#&gt; $ hits.kg        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Viruses", "Viruses", "Viruses", "Viruses", "Viruses", …</span></span>
<span><span class="co">#&gt; $ hits.ndom      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1…</span></span>
<span><span class="co">#&gt; $ hits.extlink   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "http://www.uniprot.org/uniprot/VEMP_SARS2", "http://ww…</span></span>
<span><span class="co">#&gt; $ hits.acc2      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "P0DTC4", "Q3LZW9", "P59637", "Q3I5J3", "Q0Q473", "Q0Q4…</span></span>
<span><span class="co">#&gt; $ hits.taxid     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2697049", "442736", "694009", "349344", "389167", "389…</span></span>
<span><span class="co">#&gt; $ hits.acc       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "VEMP_SARS2", "VEMP_BCHK3", "VEMP_SARS", "VEMP_BCRP3", …</span></span>
<span><span class="co">#&gt; $ hits.taxlink   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "http://www.uniprot.org/taxonomy/", "http://www.uniprot…</span></span>
<span><span class="co">#&gt; $ hits.desc      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Envelope small membrane protein", "Envelope small memb…</span></span>
<span><span class="co">#&gt; $ hits.pvalue    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> -108.52911, -100.14785, -100.14785, -100.14785, -98.919…</span></span>
<span><span class="co">#&gt; $ hits.flags     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3…</span></span>
<span><span class="co">#&gt; $ hits.nregions  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1…</span></span>
<span><span class="co">#&gt; $ hits.niseqs    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16,…</span></span>
<span><span class="co">#&gt; $ hits.name      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "VEMP_SARS2", "VEMP_BCHK3", "VEMP_SARS", "VEMP_BCRP3", …</span></span>
<span><span class="co">#&gt; $ hits.species   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Severe acute respiratory syndrome coronavirus 2", "Bat…</span></span>
<span><span class="co">#&gt; $ hits.score     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "145.7", "134.2", "134.2", "134.2", "132.5", "51.2", "5…</span></span>
<span><span class="co">#&gt; $ hits.bias      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "7.1", "6.4", "6.4", "6.4", "6.2", "3.8", "3.8", "2.9",…</span></span>
<span><span class="co">#&gt; $ hits.sindex    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "3311831", "3411115", "3412080", "3410610", "3410655", …</span></span>
<span><span class="co">#&gt; $ hits.nincluded <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1…</span></span>
<span><span class="co">#&gt; $ hits.domains   <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> [[1, "VEMP_SARS2", 75, "", 1, "&gt;Query", 145.53, 4.6e-4…</span></span>
<span><span class="co">#&gt; $ hits.pdbs      <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> ["7k3g_C", "7m4r_C", "7ntk_H", "7k3g_D", "7ntk_C", "7k…</span></span>
<span><span class="co">#&gt; $ hits.evalue    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0.0e+00, 0.0e+00, 0.0e+00, 0.0e+00, 0.0e+00, 0.0e+00, 7…</span></span>
<span><span class="co">#&gt; $ hits.nreported <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1…</span></span>
<span><span class="co">#&gt; $ hits.archindex <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "88919621081425 2697049", "88919621081425 442736", "889…</span></span></code></pre></div>
<p>To get a detailed description of each variable, please read the documentation of <code><a href="../reference/phmmer_2abl.html">?phmmer_2abl</a></code> (our example data set).</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">?</span><span class="va">phmmer_2abl</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="data-preprocessing">Data preprocessing<a class="anchor" aria-label="anchor" href="#data-preprocessing"></a>
</h4>
<div class="section level5">
<h5 id="e-values">E-values<a class="anchor" aria-label="anchor" href="#e-values"></a>
</h5>
<p>First, we’ll begin by studying the e-values obtained. There are several types of e-values, some referring to the sequence and others referring to the domains. But what do we mean by domains? According to the HMMER manual:</p>
<blockquote>
<p>A target sequence that reaches this point is very likely to contain one or more significant matches to the profile. These matches are referred to as “<em>domains</em>”, since the main use of HMMER has historically been to match profile HMMs from protein domain databases like Pfam, and one of HMMER’s strengths is to be able to cleanly parse a multidomain target sequence into its multiple nonoverlapping hits to the same domain model.</p>
</blockquote>
<p>It is of special interest to study carefully the domain <code>E-value</code> when it happens that, although the full sequence <code>E-value</code> is good, none of the domain <code>E-value</code> reaches the significance threshold. In that case, what is happening is that those weak hits, none of which is good enough on its own, add up to raise the sequence to a high score. This may happen because the sequence contains several weak homologous domains, or it may contain a repetitive sequence that is hitting by chance.</p>
<p>HMMERutils has a function to display all the <code>E-values</code> of the sequences. This function <code>hmmer_evalues_cleveland_dot_plot</code>. In this plot, the <span class="math inline">\(−log(\text{E−value})\)</span> is represented on the <span class="math inline">\(x\)</span> axis, and on the <span class="math inline">\(y\)</span> axis each of the identified sequences is represented one at a time. The green dots represent the E-values of the full sequence and the red dots the E-values of each of the domains.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hmmer_evalues_cleveland_dot_plot.html">hmmer_evalues_cleveland_dot_plot</a></span><span class="op">(</span></span>
<span>    <span class="va">hmmer.VEMP_SARS</span>,</span>
<span>    threshold <span class="op">=</span> <span class="fl">10</span><span class="op">^</span><span class="op">-</span><span class="fl">5</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="HMMERutils_files/figure-html/e-values-1.png" width="1458.33333333333"></p>
<p>By looking at it, you can see at a glance, first, how many domains have been identified for each sequence, and then how these contribute to the E-value of the full sequence and how significant they are relative to the full sequence. In this case, as can be seen, only one domain per sequence has been found (probably because they are very small).</p>
<p>As we didn’t identify any “red-flag”, let’s filter our data taking into account only the hits sequence E-value. We can use the HMMERutils function <code><a href="../reference/filter_hmmer.html">filter_hmmer()</a></code> to achieve this.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span></span>
<span>    <span class="va">hmmer.VEMP_SARS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_hmmer.html">filter_hmmer</a></span><span class="op">(</span><span class="va">hmmer.VEMP_SARS</span>, threshold <span class="op">=</span> <span class="fl">10</span><span class="op">^</span><span class="op">-</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt;    algorithm                                 uuid stats.page stats.nhits</span></span>
<span><span class="co">#&gt; 1     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 2     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 3     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 4     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 5     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 6     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 7     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 8     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 9     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 10    phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 11    phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 12    phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 13    phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 14    phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 15    phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt;    stats.elapsed stats.Z stats.Z_setby stats.n_past_msv stats.unpacked</span></span>
<span><span class="co">#&gt; 1           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 2           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 3           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 4           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 5           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 6           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 7           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 8           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 9           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 10          0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 11          0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 12          0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 13          0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 14          0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 15          0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt;    stats.nseqs stats.user stats.domZ_setby stats.n_past_bias stats.sys</span></span>
<span><span class="co">#&gt; 1       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 2       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 3       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 4       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 5       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 6       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 7       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 8       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 9       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 10      565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 11      565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 12      565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 13      565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 14      565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 15      565928          0                0             14018         0</span></span>
<span><span class="co">#&gt;    stats.n_past_fwd stats.total stats.nmodels stats.nincluded stats.n_past_vit</span></span>
<span><span class="co">#&gt; 1                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 2                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 3                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 4                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 5                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 6                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 7                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 8                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 9                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 10               60           1             1              35              902</span></span>
<span><span class="co">#&gt; 11               60           1             1              35              902</span></span>
<span><span class="co">#&gt; 12               60           1             1              35              902</span></span>
<span><span class="co">#&gt; 13               60           1             1              35              902</span></span>
<span><span class="co">#&gt; 14               60           1             1              35              902</span></span>
<span><span class="co">#&gt; 15               60           1             1              35              902</span></span>
<span><span class="co">#&gt;    stats.nreported stats.domZ hits.archScore      hits.ph  hits.arch hits.kg</span></span>
<span><span class="co">#&gt; 1               43         43              1 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 2               43         43              1 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 3               43         43              1 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 4               43         43              1 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 5               43         43              1 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 6               43         43              2 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 7               43         43              2 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 8               43         43              2 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 9               43         43              0 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 10              43         43              0 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 11              43         43              0 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 12              43         43              0 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 13              43         43              2 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 14              43         43              0 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 15              43         43              0 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt;    hits.ndom                              hits.extlink hits.acc2 hits.taxid</span></span>
<span><span class="co">#&gt; 1          1 http://www.uniprot.org/uniprot/VEMP_SARS2    P0DTC4    2697049</span></span>
<span><span class="co">#&gt; 2          1 http://www.uniprot.org/uniprot/VEMP_BCHK3    Q3LZW9     442736</span></span>
<span><span class="co">#&gt; 3          1  http://www.uniprot.org/uniprot/VEMP_SARS    P59637     694009</span></span>
<span><span class="co">#&gt; 4          1 http://www.uniprot.org/uniprot/VEMP_BCRP3    Q3I5J3     349344</span></span>
<span><span class="co">#&gt; 5          1 http://www.uniprot.org/uniprot/VEMP_BC279    Q0Q473     389167</span></span>
<span><span class="co">#&gt; 6          1 http://www.uniprot.org/uniprot/VEMP_BC133    Q0Q4E8     389230</span></span>
<span><span class="co">#&gt; 7          1 http://www.uniprot.org/uniprot/VEMP_BCHK4    A3EX99     694007</span></span>
<span><span class="co">#&gt; 8          1 http://www.uniprot.org/uniprot/VEMP_MERS1    K9N5R3    1263720</span></span>
<span><span class="co">#&gt; 9          1 http://www.uniprot.org/uniprot/VEMP_CVPPU    P09048      11151</span></span>
<span><span class="co">#&gt; 10         1 http://www.uniprot.org/uniprot/VEMP_CVHN5    Q0ZME5     443241</span></span>
<span><span class="co">#&gt; 11         1 http://www.uniprot.org/uniprot/VEMP_CVHN2    Q14EA8     443240</span></span>
<span><span class="co">#&gt; 12         1 http://www.uniprot.org/uniprot/VEMP_CVCAI    P36696      36391</span></span>
<span><span class="co">#&gt; 13         1 http://www.uniprot.org/uniprot/VEMP_BCHK5    A3EXD5     694008</span></span>
<span><span class="co">#&gt; 14         1 http://www.uniprot.org/uniprot/VEMP_CVPR8    P69611      33736</span></span>
<span><span class="co">#&gt; 15         1 http://www.uniprot.org/uniprot/VEMP_CVPRM    P69610      11148</span></span>
<span><span class="co">#&gt;      hits.acc                     hits.taxlink                       hits.desc</span></span>
<span><span class="co">#&gt; 1  VEMP_SARS2 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 2  VEMP_BCHK3 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 3   VEMP_SARS http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 4  VEMP_BCRP3 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 5  VEMP_BC279 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 6  VEMP_BC133 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 7  VEMP_BCHK4 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 8  VEMP_MERS1 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 9  VEMP_CVPPU http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 10 VEMP_CVHN5 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 11 VEMP_CVHN2 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 12 VEMP_CVCAI http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 13 VEMP_BCHK5 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 14 VEMP_CVPR8 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 15 VEMP_CVPRM http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt;    hits.pvalue hits.flags hits.nregions hits.niseqs  hits.name</span></span>
<span><span class="co">#&gt; 1   -108.52911          3             1          16 VEMP_SARS2</span></span>
<span><span class="co">#&gt; 2   -100.14785          3             1          16 VEMP_BCHK3</span></span>
<span><span class="co">#&gt; 3   -100.14785          3             1          16  VEMP_SARS</span></span>
<span><span class="co">#&gt; 4   -100.14785          3             1          16 VEMP_BCRP3</span></span>
<span><span class="co">#&gt; 5    -98.91970          3             1          16 VEMP_BC279</span></span>
<span><span class="co">#&gt; 6    -39.57741          3             1          16 VEMP_BC133</span></span>
<span><span class="co">#&gt; 7    -38.93211          3             1          16 VEMP_BCHK4</span></span>
<span><span class="co">#&gt; 8    -30.29100          3             1          16 VEMP_MERS1</span></span>
<span><span class="co">#&gt; 9    -29.06330          3             1          16 VEMP_CVPPU</span></span>
<span><span class="co">#&gt; 10   -28.90947          3             1          16 VEMP_CVHN5</span></span>
<span><span class="co">#&gt; 11   -28.90947          3             1          16 VEMP_CVHN2</span></span>
<span><span class="co">#&gt; 12   -27.84353          3             1          16 VEMP_CVCAI</span></span>
<span><span class="co">#&gt; 13   -27.36318          3             1          16 VEMP_BCHK5</span></span>
<span><span class="co">#&gt; 14   -26.76532          3             1          16 VEMP_CVPR8</span></span>
<span><span class="co">#&gt; 15   -26.76532          3             1          16 VEMP_CVPRM</span></span>
<span><span class="co">#&gt;                                                                                     hits.species</span></span>
<span><span class="co">#&gt; 1                                                Severe acute respiratory syndrome coronavirus 2</span></span>
<span><span class="co">#&gt; 2                                                                           Bat coronavirus HKU3</span></span>
<span><span class="co">#&gt; 3                                                  Severe acute respiratory syndrome coronavirus</span></span>
<span><span class="co">#&gt; 4                                                                       Bat coronavirus Rp3/2004</span></span>
<span><span class="co">#&gt; 5                                                                               Bat CoV 279/2005</span></span>
<span><span class="co">#&gt; 6                                                               Bat coronavirus (BtCoV/133/2005)</span></span>
<span><span class="co">#&gt; 7                                                                           Bat coronavirus HKU4</span></span>
<span><span class="co">#&gt; 8  Middle East respiratory syndrome-related coronavirus (isolate United Kingdom/H123990006/2012)</span></span>
<span><span class="co">#&gt; 9                              Porcine transmissible gastroenteritis coronavirus (strain Purdue)</span></span>
<span><span class="co">#&gt; 10                                                           Human coronavirus HKU1 (isolate N5)</span></span>
<span><span class="co">#&gt; 11                                                           Human coronavirus HKU1 (isolate N2)</span></span>
<span><span class="co">#&gt; 12                                                          Canine coronavirus (strain Insavc-1)</span></span>
<span><span class="co">#&gt; 13                                                                          Bat coronavirus HKU5</span></span>
<span><span class="co">#&gt; 14                          Porcine respiratory coronavirus (strain 86/137004 / isolate British)</span></span>
<span><span class="co">#&gt; 15                                                  Porcine respiratory coronavirus (strain RM4)</span></span>
<span><span class="co">#&gt;    hits.score hits.bias hits.sindex hits.nincluded</span></span>
<span><span class="co">#&gt; 1       145.7       7.1     3311831              1</span></span>
<span><span class="co">#&gt; 2       134.2       6.4     3411115              1</span></span>
<span><span class="co">#&gt; 3       134.2       6.4     3412080              1</span></span>
<span><span class="co">#&gt; 4       134.2       6.4     3410610              1</span></span>
<span><span class="co">#&gt; 5       132.5       6.2     3410655              1</span></span>
<span><span class="co">#&gt; 6        51.2       3.8     3410660              1</span></span>
<span><span class="co">#&gt; 7        50.3       3.8     3411831              1</span></span>
<span><span class="co">#&gt; 8        38.4       2.9     3140808              1</span></span>
<span><span class="co">#&gt; 9        36.7       2.0     3128822              1</span></span>
<span><span class="co">#&gt; 10       36.5       1.9     3411155              1</span></span>
<span><span class="co">#&gt; 11       36.5       1.9     3411146              1</span></span>
<span><span class="co">#&gt; 12       35.1       1.4     3138702              1</span></span>
<span><span class="co">#&gt; 13       34.4       4.0     3411906              1</span></span>
<span><span class="co">#&gt; 14       33.6       1.7     3138691              1</span></span>
<span><span class="co">#&gt; 15       33.6       1.7     3128508              1</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   hits.domains</span></span>
<span><span class="co">#&gt; 1                                      1, VEMP_SARS2, 75, , 1, &gt;Query, 145.530029296875, 4.6e-42, 75, 1, 75, 7.10, 1, 3.5e-46, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnssrvpdllv, 1, 75, 1, , 3311831, 75, , , 0.99, MYSFVSEETGTLIVNSVLLFLAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPSFYVYSRVKNLNSSRVPDLLV, 75, 1, 8************************************************************************97, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnssrvpdllv, 75, 1, , 75, , Envelope small membrane protein, VEMP_SARS2, 1, 75</span></span>
<span><span class="co">#&gt; 2               1, VEMP_BCHK3, 72, , 1, &gt;Query, 134.040939331055, 2.0e-38, 76, 0.986666666666667, 76, 6.37, 1, 1.5e-42, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkp+ yvysrvknlnss+ vpdllv, 1, 76, 1, , 3411115, 76, , , 0.97, MYSFVSEETGTLIVNSVLLFLAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPTVYVYSRVKNLNSSEgVPDLLV, 75, 0.96, 8*****************************************************************98659***97, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnssr.vpdllv, 75, 1, , 74, , Envelope small membrane protein, VEMP_BCHK3, 1, 76</span></span>
<span><span class="co">#&gt; 3                 1, VEMP_SARS, 72, , 1, &gt;Query, 134.040939331055, 2.0e-38, 76, 0.986666666666667, 76, 6.37, 1, 1.5e-42, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkp+ yvysrvknlnss+ vpdllv, 1, 76, 1, , 3412080, 76, , , 0.97, MYSFVSEETGTLIVNSVLLFLAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPTVYVYSRVKNLNSSEgVPDLLV, 75, 0.96, 8*****************************************************************98659***97, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnssr.vpdllv, 75, 1, , 74, , Envelope small membrane protein, VEMP_SARS, 1, 76</span></span>
<span><span class="co">#&gt; 4               1, VEMP_BCRP3, 72, , 1, &gt;Query, 134.040939331055, 2.0e-38, 76, 0.986666666666667, 76, 6.37, 1, 1.5e-42, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkp+ yvysrvknlnss+ vpdllv, 1, 76, 1, , 3410610, 76, , , 0.97, MYSFVSEETGTLIVNSVLLFLAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPTVYVYSRVKNLNSSEgVPDLLV, 75, 0.96, 8*****************************************************************98659***97, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnssr.vpdllv, 75, 1, , 74, , Envelope small membrane protein, VEMP_BCRP3, 1, 76</span></span>
<span><span class="co">#&gt; 5  1, VEMP_BC279, 71, , 1, &gt;Query, 132.357696533203, 6.8e-38, 76, 0.986666666666667, 76, 6.22, 1, 5.2e-42, mysfvseetgtlivnsvllf+afvvfllvtlailtalrlcayccnivnvslvkp+ yvysrvknlnss+ vpdllv, 1, 76, 1, , 3410655, 76, , , 0.97, MYSFVSEETGTLIVNSVLLFFAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPTVYVYSRVKNLNSSEgVPDLLV, 75, 0.946666666666667, 8*****************************************************************98659***97, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnssr.vpdllv, 75, 1, , 74, , Envelope small membrane protein, VEMP_BC279, 1, 76</span></span>
<span><span class="co">#&gt; 6                                                                     1, VEMP_BC133, 30, , 1, &gt;Query, 50.9613990783691, 4.2e-12, 64, 0.671875, 64, 3.81, 1, 3.2e-16, m  fv e+ gt+ivn  +l +   + llv la+lta+rlc  c++ vn  l  p+fy+y+  +n, 1, 82, 1, , 3410660, 71, , , 0.95, MLPFVHEQIGTIIVNFFILTVVCAITLLVCLAVLTAIRLCVQCASGVNTLLFVPAFYIYNTGRN, 64, 0.46875, 5679*******************************************************98877, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvkn, 75, 1, , 43, , Envelope small membrane protein, VEMP_BC133, 1, 64</span></span>
<span><span class="co">#&gt; 7                                                                     1, VEMP_BCHK4, 30, , 1, &gt;Query, 50.0769119262695, 8.0e-12, 64, 0.671875, 64, 3.78, 1, 6.1e-16, m  fv e+ gt+ivn  +l +   + l+v lailta+rlc  c++ vn  l  p+fy+y+  +n, 1, 82, 1, , 3411831, 71, , , 0.95, MLPFVHEQIGTIIVNFFILTVVCAITLVVCLAILTAIRLCVQCASGVNTLLFVPAFYIYNTGRN, 64, 0.46875, 5679*******************************************************98877, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvkn, 75, 1, , 43, , Envelope small membrane protein, VEMP_BCHK4, 1, 64</span></span>
<span><span class="co">#&gt; 8                                                      1, VEMP_MERS1, 25, , 1, &gt;Query, 38.2747611999512, 4.4e-08, 63, 0.603174603174603, 63, 2.90, 1, 3.4e-12, m  fv e  g +ivn  ++ +   + llv +a+lta rlc  c    n  lv+p++y+y+  +, 1, 82, 1, , 3140808, 73, , , 0.93, MLPFVQERIGLFIVNFFIFTVVCAITLLVCMAFLTATRLCVQCMTGFNTLLVQPALYLYNTGR, 63, 0.396825396825397, 5679************999999*************************************8755, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvk, 75, 1, , 38, , Envelope small membrane protein, VEMP_MERS1, 1, 63</span></span>
<span><span class="co">#&gt; 9                                                                        1, VEMP_CVPPU, 15, , 1, &gt;Query, 36.5278587341309, 1.6e-07, 69, 0.649122807017544, 69, 1.98, 6, 1.2e-11,    +++n ++ fl +++++l+++a+l  ++lc  ccn+    ++ p+ + y   kn+ , 9, 82, 1, , 3128822, 77, , , 0.90, N-GMVINIIFWFLLIIILILLSIALLNIIKLCMVCCNLGRTVIIVPAQHAYDAYKNFM, 66, 0.263157894736842, 4.4789***********************************************99974, tgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknln, 75, 13, , 37, , Envelope small membrane protein, VEMP_CVPPU, 13, 58</span></span>
<span><span class="co">#&gt; 10                                                                                      1, VEMP_CVHN5, 19, , 1, &gt;Query, 36.2906875610352, 1.9e-07, 68, 0.611111111111111, 68, 1.90, 3, 1.4e-11,    +l l+ v+fl+  +a+l  ++lc   c + n+ ++ ps yvy+r + l  s, 15, 82, 1, , 3411155, 75, , , 0.86, QIFFLVLSCVIFLIFVVALLATIKLCIQICGFCNIFIISPSAYVYNRGRQLYKS, 68, 0.351851851851852, 455788999**************************************9998766, nsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnss, 75, 15, , 33, , Envelope small membrane protein, VEMP_CVHN5, 15, 54</span></span>
<span><span class="co">#&gt; 11                                                                                      1, VEMP_CVHN2, 19, , 1, &gt;Query, 36.2906875610352, 1.9e-07, 68, 0.611111111111111, 68, 1.90, 3, 1.4e-11,    +l l+ v+fl+  +a+l  ++lc   c + n+ ++ ps yvy+r + l  s, 15, 82, 1, , 3411146, 75, , , 0.86, QIFFLVLSCVIFLIFVVALLATIKLCIQICGFCNIFIISPSAYVYNRGRQLYKS, 68, 0.351851851851852, 455788999**************************************9998766, nsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnss, 75, 15, , 33, , Envelope small membrane protein, VEMP_CVHN2, 15, 54</span></span>
<span><span class="co">#&gt; 12                                                           1, VEMP_CVCAI, 16, , 1, &gt;Query, 34.8620529174805, 5.3e-07, 72, 0.633333333333333, 72, 1.40, 6, 4.1e-11,  g ++++ ++ fl +++++l ++a+l  ++lc  ccn+    ++ p+ + y   kn+   r, 9, 82, 1, , 3138702, 79, , , 0.88, NG-MVISIIFWFLLIIILILFSIALLNIIKLCMVCCNLGRTVIIVPARHAYDAYKNFMQIR, 69, 0.266666666666667, 44.67889999********************************************987655, tgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnssr, 75, 13, , 38, , Envelope small membrane protein, VEMP_CVCAI, 13, 61</span></span>
<span><span class="co">#&gt; 13                                                               1, VEMP_BCHK5, 24, , 1, &gt;Query, 34.1980476379395, 8.7e-07, 61, 0.60655737704918, 61, 4.02, 1, 6.6e-11, m  fv e+ g +ivn  +l +   v l+v lailta+rlc  c +  +  +  p+ ++y+ , 1, 82, 1, , 3411906, 68, , , 0.94, MLPFVQEQIGAFIVNFFILSVVCAVTLVVCLAILTAIRLCVQCVSGCHTLVFLPAVHIYNT, 61, 0.39344262295082, 5679********************************************99999****9975, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysr, 75, 1, , 37, , Envelope small membrane protein, VEMP_BCHK5, 1, 61</span></span>
<span><span class="co">#&gt; 14                                                                       1, VEMP_CVPR8, 15, , 1, &gt;Query, 33.3970336914062, 1.6e-06, 69, 0.649122807017544, 69, 1.73, 6, 1.2e-10,  g ++++ ++ fl +++++l+++a+l  ++lc  ccn+    ++ p  + y   kn+ , 9, 82, 1, , 3138691, 78, , , 0.88, NG-MVISIIFWFLLIIILILLSIALLNIIKLCMVCCNLGRTVIIVPVQHAYDAYKNFM, 66, 0.263157894736842, 44.67889999*****************************************999974, tgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknln, 75, 13, , 37, , Envelope small membrane protein, VEMP_CVPR8, 13, 58</span></span>
<span><span class="co">#&gt; 15                                                                       1, VEMP_CVPRM, 15, , 1, &gt;Query, 33.3970336914062, 1.6e-06, 69, 0.649122807017544, 69, 1.73, 6, 1.2e-10,  g ++++ ++ fl +++++l+++a+l  ++lc  ccn+    ++ p  + y   kn+ , 9, 82, 1, , 3128508, 78, , , 0.88, NG-MVISIIFWFLLIIILILLSIALLNIIKLCMVCCNLGRTVIIVPVQHAYDAYKNFM, 66, 0.263157894736842, 44.67889999*****************************************999974, tgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknln, 75, 13, , 37, , Envelope small membrane protein, VEMP_CVPRM, 13, 58</span></span>
<span><span class="co">#&gt;                                                                         hits.pdbs</span></span>
<span><span class="co">#&gt; 1  7k3g_C, 7m4r_C, 7ntk_H, 7k3g_D, 7ntk_C, 7k3g_A, 7k3g_E, 7k3g_B, 7ntk_G, 7ntk_E</span></span>
<span><span class="co">#&gt; 2                                                                            NULL</span></span>
<span><span class="co">#&gt; 3  5x29_E, 7ntj_G, 5x29_D, 5x29_A, 5xes_A, 2mm4_A, 5x29_B, 7ntj_C, 5xer_A, 5x29_C</span></span>
<span><span class="co">#&gt; 4                                                                            NULL</span></span>
<span><span class="co">#&gt; 5                                                                            NULL</span></span>
<span><span class="co">#&gt; 6                                                                            NULL</span></span>
<span><span class="co">#&gt; 7                                                                            NULL</span></span>
<span><span class="co">#&gt; 8                                                                            NULL</span></span>
<span><span class="co">#&gt; 9                                                                            NULL</span></span>
<span><span class="co">#&gt; 10                                                                           NULL</span></span>
<span><span class="co">#&gt; 11                                                                           NULL</span></span>
<span><span class="co">#&gt; 12                                                                           NULL</span></span>
<span><span class="co">#&gt; 13                                                                           NULL</span></span>
<span><span class="co">#&gt; 14                                                                           NULL</span></span>
<span><span class="co">#&gt; 15                                                                           NULL</span></span>
<span><span class="co">#&gt;    hits.evalue hits.nreported         hits.archindex</span></span>
<span><span class="co">#&gt; 1        0e+00              1 88919621081425 2697049</span></span>
<span><span class="co">#&gt; 2        0e+00              1  88919621081425 442736</span></span>
<span><span class="co">#&gt; 3        0e+00              1  88919621081425 694009</span></span>
<span><span class="co">#&gt; 4        0e+00              1  88919621081425 349344</span></span>
<span><span class="co">#&gt; 5        0e+00              1  88919621081425 389167</span></span>
<span><span class="co">#&gt; 6        0e+00              1  88919621081425 389230</span></span>
<span><span class="co">#&gt; 7        7e-12              1  88919621081425 694007</span></span>
<span><span class="co">#&gt; 8        4e-08              1 88919621081425 1263720</span></span>
<span><span class="co">#&gt; 9        0e+00              1   88919621081425 11151</span></span>
<span><span class="co">#&gt; 10       0e+00              1  88919621081425 443241</span></span>
<span><span class="co">#&gt; 11       0e+00              1  88919621081425 443240</span></span>
<span><span class="co">#&gt; 12       0e+00              1   88919621081425 36391</span></span>
<span><span class="co">#&gt; 13       1e-06              1  88919621081425 694008</span></span>
<span><span class="co">#&gt; 14       1e-06              1   88919621081425 33736</span></span>
<span><span class="co">#&gt; 15       1e-06              1   88919621081425 11148</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="adding-protein-sequences">Adding protein sequences<a class="anchor" aria-label="anchor" href="#adding-protein-sequences"></a>
</h5>
<p>To download the sequences from HMMER and add them to our DataFrame, you can use the <code><a href="../reference/add_sequences_to_hmmer_tbl.html">add_sequences_to_hmmer_tbl()</a></code>. Notice that sequences associated with the <em>UUID</em> will be available in HMMER only temporarily.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span></span>
<span>    <span class="va">hmmer.VEMP_SARS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/add_sequences_to_hmmer_tbl.html">add_sequences_to_hmmer_tbl</a></span><span class="op">(</span><span class="va">hmmer.VEMP_SARS</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt;    algorithm                                 uuid stats.page stats.nhits</span></span>
<span><span class="co">#&gt; 1     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 2     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 3     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 4     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 5     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 6     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 7     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 8     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 9     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 10    phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 11    phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 12    phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 13    phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 14    phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 15    phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt;    stats.elapsed stats.Z stats.Z_setby stats.n_past_msv stats.unpacked</span></span>
<span><span class="co">#&gt; 1           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 2           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 3           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 4           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 5           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 6           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 7           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 8           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 9           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 10          0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 11          0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 12          0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 13          0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 14          0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 15          0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt;    stats.nseqs stats.user stats.domZ_setby stats.n_past_bias stats.sys</span></span>
<span><span class="co">#&gt; 1       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 2       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 3       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 4       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 5       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 6       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 7       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 8       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 9       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 10      565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 11      565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 12      565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 13      565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 14      565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 15      565928          0                0             14018         0</span></span>
<span><span class="co">#&gt;    stats.n_past_fwd stats.total stats.nmodels stats.nincluded stats.n_past_vit</span></span>
<span><span class="co">#&gt; 1                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 2                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 3                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 4                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 5                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 6                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 7                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 8                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 9                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 10               60           1             1              35              902</span></span>
<span><span class="co">#&gt; 11               60           1             1              35              902</span></span>
<span><span class="co">#&gt; 12               60           1             1              35              902</span></span>
<span><span class="co">#&gt; 13               60           1             1              35              902</span></span>
<span><span class="co">#&gt; 14               60           1             1              35              902</span></span>
<span><span class="co">#&gt; 15               60           1             1              35              902</span></span>
<span><span class="co">#&gt;    stats.nreported stats.domZ hits.archScore      hits.ph  hits.arch hits.kg</span></span>
<span><span class="co">#&gt; 1               43         43              1 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 2               43         43              1 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 3               43         43              1 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 4               43         43              1 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 5               43         43              1 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 6               43         43              2 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 7               43         43              2 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 8               43         43              2 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 9               43         43              0 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 10              43         43              0 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 11              43         43              0 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 12              43         43              0 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 13              43         43              2 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 14              43         43              0 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 15              43         43              0 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt;    hits.ndom                              hits.extlink hits.acc2 hits.taxid</span></span>
<span><span class="co">#&gt; 1          1 http://www.uniprot.org/uniprot/VEMP_SARS2    P0DTC4    2697049</span></span>
<span><span class="co">#&gt; 2          1 http://www.uniprot.org/uniprot/VEMP_BCHK3    Q3LZW9     442736</span></span>
<span><span class="co">#&gt; 3          1  http://www.uniprot.org/uniprot/VEMP_SARS    P59637     694009</span></span>
<span><span class="co">#&gt; 4          1 http://www.uniprot.org/uniprot/VEMP_BCRP3    Q3I5J3     349344</span></span>
<span><span class="co">#&gt; 5          1 http://www.uniprot.org/uniprot/VEMP_BC279    Q0Q473     389167</span></span>
<span><span class="co">#&gt; 6          1 http://www.uniprot.org/uniprot/VEMP_BC133    Q0Q4E8     389230</span></span>
<span><span class="co">#&gt; 7          1 http://www.uniprot.org/uniprot/VEMP_BCHK4    A3EX99     694007</span></span>
<span><span class="co">#&gt; 8          1 http://www.uniprot.org/uniprot/VEMP_MERS1    K9N5R3    1263720</span></span>
<span><span class="co">#&gt; 9          1 http://www.uniprot.org/uniprot/VEMP_CVPPU    P09048      11151</span></span>
<span><span class="co">#&gt; 10         1 http://www.uniprot.org/uniprot/VEMP_CVHN5    Q0ZME5     443241</span></span>
<span><span class="co">#&gt; 11         1 http://www.uniprot.org/uniprot/VEMP_CVHN2    Q14EA8     443240</span></span>
<span><span class="co">#&gt; 12         1 http://www.uniprot.org/uniprot/VEMP_CVCAI    P36696      36391</span></span>
<span><span class="co">#&gt; 13         1 http://www.uniprot.org/uniprot/VEMP_BCHK5    A3EXD5     694008</span></span>
<span><span class="co">#&gt; 14         1 http://www.uniprot.org/uniprot/VEMP_CVPR8    P69611      33736</span></span>
<span><span class="co">#&gt; 15         1 http://www.uniprot.org/uniprot/VEMP_CVPRM    P69610      11148</span></span>
<span><span class="co">#&gt;      hits.acc                     hits.taxlink                       hits.desc</span></span>
<span><span class="co">#&gt; 1  VEMP_SARS2 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 2  VEMP_BCHK3 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 3   VEMP_SARS http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 4  VEMP_BCRP3 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 5  VEMP_BC279 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 6  VEMP_BC133 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 7  VEMP_BCHK4 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 8  VEMP_MERS1 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 9  VEMP_CVPPU http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 10 VEMP_CVHN5 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 11 VEMP_CVHN2 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 12 VEMP_CVCAI http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 13 VEMP_BCHK5 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 14 VEMP_CVPR8 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 15 VEMP_CVPRM http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt;    hits.pvalue hits.flags hits.nregions hits.niseqs  hits.name</span></span>
<span><span class="co">#&gt; 1   -108.52911          3             1          16 VEMP_SARS2</span></span>
<span><span class="co">#&gt; 2   -100.14785          3             1          16 VEMP_BCHK3</span></span>
<span><span class="co">#&gt; 3   -100.14785          3             1          16  VEMP_SARS</span></span>
<span><span class="co">#&gt; 4   -100.14785          3             1          16 VEMP_BCRP3</span></span>
<span><span class="co">#&gt; 5    -98.91970          3             1          16 VEMP_BC279</span></span>
<span><span class="co">#&gt; 6    -39.57741          3             1          16 VEMP_BC133</span></span>
<span><span class="co">#&gt; 7    -38.93211          3             1          16 VEMP_BCHK4</span></span>
<span><span class="co">#&gt; 8    -30.29100          3             1          16 VEMP_MERS1</span></span>
<span><span class="co">#&gt; 9    -29.06330          3             1          16 VEMP_CVPPU</span></span>
<span><span class="co">#&gt; 10   -28.90947          3             1          16 VEMP_CVHN5</span></span>
<span><span class="co">#&gt; 11   -28.90947          3             1          16 VEMP_CVHN2</span></span>
<span><span class="co">#&gt; 12   -27.84353          3             1          16 VEMP_CVCAI</span></span>
<span><span class="co">#&gt; 13   -27.36318          3             1          16 VEMP_BCHK5</span></span>
<span><span class="co">#&gt; 14   -26.76532          3             1          16 VEMP_CVPR8</span></span>
<span><span class="co">#&gt; 15   -26.76532          3             1          16 VEMP_CVPRM</span></span>
<span><span class="co">#&gt;                                                                                     hits.species</span></span>
<span><span class="co">#&gt; 1                                                Severe acute respiratory syndrome coronavirus 2</span></span>
<span><span class="co">#&gt; 2                                                                           Bat coronavirus HKU3</span></span>
<span><span class="co">#&gt; 3                                                  Severe acute respiratory syndrome coronavirus</span></span>
<span><span class="co">#&gt; 4                                                                       Bat coronavirus Rp3/2004</span></span>
<span><span class="co">#&gt; 5                                                                               Bat CoV 279/2005</span></span>
<span><span class="co">#&gt; 6                                                               Bat coronavirus (BtCoV/133/2005)</span></span>
<span><span class="co">#&gt; 7                                                                           Bat coronavirus HKU4</span></span>
<span><span class="co">#&gt; 8  Middle East respiratory syndrome-related coronavirus (isolate United Kingdom/H123990006/2012)</span></span>
<span><span class="co">#&gt; 9                              Porcine transmissible gastroenteritis coronavirus (strain Purdue)</span></span>
<span><span class="co">#&gt; 10                                                           Human coronavirus HKU1 (isolate N5)</span></span>
<span><span class="co">#&gt; 11                                                           Human coronavirus HKU1 (isolate N2)</span></span>
<span><span class="co">#&gt; 12                                                          Canine coronavirus (strain Insavc-1)</span></span>
<span><span class="co">#&gt; 13                                                                          Bat coronavirus HKU5</span></span>
<span><span class="co">#&gt; 14                          Porcine respiratory coronavirus (strain 86/137004 / isolate British)</span></span>
<span><span class="co">#&gt; 15                                                  Porcine respiratory coronavirus (strain RM4)</span></span>
<span><span class="co">#&gt;    hits.score hits.bias hits.sindex hits.nincluded</span></span>
<span><span class="co">#&gt; 1       145.7       7.1     3311831              1</span></span>
<span><span class="co">#&gt; 2       134.2       6.4     3411115              1</span></span>
<span><span class="co">#&gt; 3       134.2       6.4     3412080              1</span></span>
<span><span class="co">#&gt; 4       134.2       6.4     3410610              1</span></span>
<span><span class="co">#&gt; 5       132.5       6.2     3410655              1</span></span>
<span><span class="co">#&gt; 6        51.2       3.8     3410660              1</span></span>
<span><span class="co">#&gt; 7        50.3       3.8     3411831              1</span></span>
<span><span class="co">#&gt; 8        38.4       2.9     3140808              1</span></span>
<span><span class="co">#&gt; 9        36.7       2.0     3128822              1</span></span>
<span><span class="co">#&gt; 10       36.5       1.9     3411155              1</span></span>
<span><span class="co">#&gt; 11       36.5       1.9     3411146              1</span></span>
<span><span class="co">#&gt; 12       35.1       1.4     3138702              1</span></span>
<span><span class="co">#&gt; 13       34.4       4.0     3411906              1</span></span>
<span><span class="co">#&gt; 14       33.6       1.7     3138691              1</span></span>
<span><span class="co">#&gt; 15       33.6       1.7     3128508              1</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   hits.domains</span></span>
<span><span class="co">#&gt; 1                                      1, VEMP_SARS2, 75, , 1, &gt;Query, 145.530029296875, 4.6e-42, 75, 1, 75, 7.10, 1, 3.5e-46, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnssrvpdllv, 1, 75, 1, , 3311831, 75, , , 0.99, MYSFVSEETGTLIVNSVLLFLAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPSFYVYSRVKNLNSSRVPDLLV, 75, 1, 8************************************************************************97, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnssrvpdllv, 75, 1, , 75, , Envelope small membrane protein, VEMP_SARS2, 1, 75</span></span>
<span><span class="co">#&gt; 2               1, VEMP_BCHK3, 72, , 1, &gt;Query, 134.040939331055, 2.0e-38, 76, 0.986666666666667, 76, 6.37, 1, 1.5e-42, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkp+ yvysrvknlnss+ vpdllv, 1, 76, 1, , 3411115, 76, , , 0.97, MYSFVSEETGTLIVNSVLLFLAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPTVYVYSRVKNLNSSEgVPDLLV, 75, 0.96, 8*****************************************************************98659***97, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnssr.vpdllv, 75, 1, , 74, , Envelope small membrane protein, VEMP_BCHK3, 1, 76</span></span>
<span><span class="co">#&gt; 3                 1, VEMP_SARS, 72, , 1, &gt;Query, 134.040939331055, 2.0e-38, 76, 0.986666666666667, 76, 6.37, 1, 1.5e-42, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkp+ yvysrvknlnss+ vpdllv, 1, 76, 1, , 3412080, 76, , , 0.97, MYSFVSEETGTLIVNSVLLFLAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPTVYVYSRVKNLNSSEgVPDLLV, 75, 0.96, 8*****************************************************************98659***97, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnssr.vpdllv, 75, 1, , 74, , Envelope small membrane protein, VEMP_SARS, 1, 76</span></span>
<span><span class="co">#&gt; 4               1, VEMP_BCRP3, 72, , 1, &gt;Query, 134.040939331055, 2.0e-38, 76, 0.986666666666667, 76, 6.37, 1, 1.5e-42, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkp+ yvysrvknlnss+ vpdllv, 1, 76, 1, , 3410610, 76, , , 0.97, MYSFVSEETGTLIVNSVLLFLAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPTVYVYSRVKNLNSSEgVPDLLV, 75, 0.96, 8*****************************************************************98659***97, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnssr.vpdllv, 75, 1, , 74, , Envelope small membrane protein, VEMP_BCRP3, 1, 76</span></span>
<span><span class="co">#&gt; 5  1, VEMP_BC279, 71, , 1, &gt;Query, 132.357696533203, 6.8e-38, 76, 0.986666666666667, 76, 6.22, 1, 5.2e-42, mysfvseetgtlivnsvllf+afvvfllvtlailtalrlcayccnivnvslvkp+ yvysrvknlnss+ vpdllv, 1, 76, 1, , 3410655, 76, , , 0.97, MYSFVSEETGTLIVNSVLLFFAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPTVYVYSRVKNLNSSEgVPDLLV, 75, 0.946666666666667, 8*****************************************************************98659***97, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnssr.vpdllv, 75, 1, , 74, , Envelope small membrane protein, VEMP_BC279, 1, 76</span></span>
<span><span class="co">#&gt; 6                                                                     1, VEMP_BC133, 30, , 1, &gt;Query, 50.9613990783691, 4.2e-12, 64, 0.671875, 64, 3.81, 1, 3.2e-16, m  fv e+ gt+ivn  +l +   + llv la+lta+rlc  c++ vn  l  p+fy+y+  +n, 1, 82, 1, , 3410660, 71, , , 0.95, MLPFVHEQIGTIIVNFFILTVVCAITLLVCLAVLTAIRLCVQCASGVNTLLFVPAFYIYNTGRN, 64, 0.46875, 5679*******************************************************98877, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvkn, 75, 1, , 43, , Envelope small membrane protein, VEMP_BC133, 1, 64</span></span>
<span><span class="co">#&gt; 7                                                                     1, VEMP_BCHK4, 30, , 1, &gt;Query, 50.0769119262695, 8.0e-12, 64, 0.671875, 64, 3.78, 1, 6.1e-16, m  fv e+ gt+ivn  +l +   + l+v lailta+rlc  c++ vn  l  p+fy+y+  +n, 1, 82, 1, , 3411831, 71, , , 0.95, MLPFVHEQIGTIIVNFFILTVVCAITLVVCLAILTAIRLCVQCASGVNTLLFVPAFYIYNTGRN, 64, 0.46875, 5679*******************************************************98877, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvkn, 75, 1, , 43, , Envelope small membrane protein, VEMP_BCHK4, 1, 64</span></span>
<span><span class="co">#&gt; 8                                                      1, VEMP_MERS1, 25, , 1, &gt;Query, 38.2747611999512, 4.4e-08, 63, 0.603174603174603, 63, 2.90, 1, 3.4e-12, m  fv e  g +ivn  ++ +   + llv +a+lta rlc  c    n  lv+p++y+y+  +, 1, 82, 1, , 3140808, 73, , , 0.93, MLPFVQERIGLFIVNFFIFTVVCAITLLVCMAFLTATRLCVQCMTGFNTLLVQPALYLYNTGR, 63, 0.396825396825397, 5679************999999*************************************8755, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvk, 75, 1, , 38, , Envelope small membrane protein, VEMP_MERS1, 1, 63</span></span>
<span><span class="co">#&gt; 9                                                                        1, VEMP_CVPPU, 15, , 1, &gt;Query, 36.5278587341309, 1.6e-07, 69, 0.649122807017544, 69, 1.98, 6, 1.2e-11,    +++n ++ fl +++++l+++a+l  ++lc  ccn+    ++ p+ + y   kn+ , 9, 82, 1, , 3128822, 77, , , 0.90, N-GMVINIIFWFLLIIILILLSIALLNIIKLCMVCCNLGRTVIIVPAQHAYDAYKNFM, 66, 0.263157894736842, 4.4789***********************************************99974, tgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknln, 75, 13, , 37, , Envelope small membrane protein, VEMP_CVPPU, 13, 58</span></span>
<span><span class="co">#&gt; 10                                                                                      1, VEMP_CVHN5, 19, , 1, &gt;Query, 36.2906875610352, 1.9e-07, 68, 0.611111111111111, 68, 1.90, 3, 1.4e-11,    +l l+ v+fl+  +a+l  ++lc   c + n+ ++ ps yvy+r + l  s, 15, 82, 1, , 3411155, 75, , , 0.86, QIFFLVLSCVIFLIFVVALLATIKLCIQICGFCNIFIISPSAYVYNRGRQLYKS, 68, 0.351851851851852, 455788999**************************************9998766, nsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnss, 75, 15, , 33, , Envelope small membrane protein, VEMP_CVHN5, 15, 54</span></span>
<span><span class="co">#&gt; 11                                                                                      1, VEMP_CVHN2, 19, , 1, &gt;Query, 36.2906875610352, 1.9e-07, 68, 0.611111111111111, 68, 1.90, 3, 1.4e-11,    +l l+ v+fl+  +a+l  ++lc   c + n+ ++ ps yvy+r + l  s, 15, 82, 1, , 3411146, 75, , , 0.86, QIFFLVLSCVIFLIFVVALLATIKLCIQICGFCNIFIISPSAYVYNRGRQLYKS, 68, 0.351851851851852, 455788999**************************************9998766, nsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnss, 75, 15, , 33, , Envelope small membrane protein, VEMP_CVHN2, 15, 54</span></span>
<span><span class="co">#&gt; 12                                                           1, VEMP_CVCAI, 16, , 1, &gt;Query, 34.8620529174805, 5.3e-07, 72, 0.633333333333333, 72, 1.40, 6, 4.1e-11,  g ++++ ++ fl +++++l ++a+l  ++lc  ccn+    ++ p+ + y   kn+   r, 9, 82, 1, , 3138702, 79, , , 0.88, NG-MVISIIFWFLLIIILILFSIALLNIIKLCMVCCNLGRTVIIVPARHAYDAYKNFMQIR, 69, 0.266666666666667, 44.67889999********************************************987655, tgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnssr, 75, 13, , 38, , Envelope small membrane protein, VEMP_CVCAI, 13, 61</span></span>
<span><span class="co">#&gt; 13                                                               1, VEMP_BCHK5, 24, , 1, &gt;Query, 34.1980476379395, 8.7e-07, 61, 0.60655737704918, 61, 4.02, 1, 6.6e-11, m  fv e+ g +ivn  +l +   v l+v lailta+rlc  c +  +  +  p+ ++y+ , 1, 82, 1, , 3411906, 68, , , 0.94, MLPFVQEQIGAFIVNFFILSVVCAVTLVVCLAILTAIRLCVQCVSGCHTLVFLPAVHIYNT, 61, 0.39344262295082, 5679********************************************99999****9975, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysr, 75, 1, , 37, , Envelope small membrane protein, VEMP_BCHK5, 1, 61</span></span>
<span><span class="co">#&gt; 14                                                                       1, VEMP_CVPR8, 15, , 1, &gt;Query, 33.3970336914062, 1.6e-06, 69, 0.649122807017544, 69, 1.73, 6, 1.2e-10,  g ++++ ++ fl +++++l+++a+l  ++lc  ccn+    ++ p  + y   kn+ , 9, 82, 1, , 3138691, 78, , , 0.88, NG-MVISIIFWFLLIIILILLSIALLNIIKLCMVCCNLGRTVIIVPVQHAYDAYKNFM, 66, 0.263157894736842, 44.67889999*****************************************999974, tgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknln, 75, 13, , 37, , Envelope small membrane protein, VEMP_CVPR8, 13, 58</span></span>
<span><span class="co">#&gt; 15                                                                       1, VEMP_CVPRM, 15, , 1, &gt;Query, 33.3970336914062, 1.6e-06, 69, 0.649122807017544, 69, 1.73, 6, 1.2e-10,  g ++++ ++ fl +++++l+++a+l  ++lc  ccn+    ++ p  + y   kn+ , 9, 82, 1, , 3128508, 78, , , 0.88, NG-MVISIIFWFLLIIILILLSIALLNIIKLCMVCCNLGRTVIIVPVQHAYDAYKNFM, 66, 0.263157894736842, 44.67889999*****************************************999974, tgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknln, 75, 13, , 37, , Envelope small membrane protein, VEMP_CVPRM, 13, 58</span></span>
<span><span class="co">#&gt;                                                                         hits.pdbs</span></span>
<span><span class="co">#&gt; 1  7k3g_C, 7m4r_C, 7ntk_H, 7k3g_D, 7ntk_C, 7k3g_A, 7k3g_E, 7k3g_B, 7ntk_G, 7ntk_E</span></span>
<span><span class="co">#&gt; 2                                                                            NULL</span></span>
<span><span class="co">#&gt; 3  5x29_E, 7ntj_G, 5x29_D, 5x29_A, 5xes_A, 2mm4_A, 5x29_B, 7ntj_C, 5xer_A, 5x29_C</span></span>
<span><span class="co">#&gt; 4                                                                            NULL</span></span>
<span><span class="co">#&gt; 5                                                                            NULL</span></span>
<span><span class="co">#&gt; 6                                                                            NULL</span></span>
<span><span class="co">#&gt; 7                                                                            NULL</span></span>
<span><span class="co">#&gt; 8                                                                            NULL</span></span>
<span><span class="co">#&gt; 9                                                                            NULL</span></span>
<span><span class="co">#&gt; 10                                                                           NULL</span></span>
<span><span class="co">#&gt; 11                                                                           NULL</span></span>
<span><span class="co">#&gt; 12                                                                           NULL</span></span>
<span><span class="co">#&gt; 13                                                                           NULL</span></span>
<span><span class="co">#&gt; 14                                                                           NULL</span></span>
<span><span class="co">#&gt; 15                                                                           NULL</span></span>
<span><span class="co">#&gt;    hits.evalue hits.nreported         hits.archindex</span></span>
<span><span class="co">#&gt; 1        0e+00              1 88919621081425 2697049</span></span>
<span><span class="co">#&gt; 2        0e+00              1  88919621081425 442736</span></span>
<span><span class="co">#&gt; 3        0e+00              1  88919621081425 694009</span></span>
<span><span class="co">#&gt; 4        0e+00              1  88919621081425 349344</span></span>
<span><span class="co">#&gt; 5        0e+00              1  88919621081425 389167</span></span>
<span><span class="co">#&gt; 6        0e+00              1  88919621081425 389230</span></span>
<span><span class="co">#&gt; 7        7e-12              1  88919621081425 694007</span></span>
<span><span class="co">#&gt; 8        4e-08              1 88919621081425 1263720</span></span>
<span><span class="co">#&gt; 9        0e+00              1   88919621081425 11151</span></span>
<span><span class="co">#&gt; 10       0e+00              1  88919621081425 443241</span></span>
<span><span class="co">#&gt; 11       0e+00              1  88919621081425 443240</span></span>
<span><span class="co">#&gt; 12       0e+00              1   88919621081425 36391</span></span>
<span><span class="co">#&gt; 13       1e-06              1  88919621081425 694008</span></span>
<span><span class="co">#&gt; 14       1e-06              1   88919621081425 33736</span></span>
<span><span class="co">#&gt; 15       1e-06              1   88919621081425 11148</span></span>
<span><span class="co">#&gt;                                                                        hits.fullfasta</span></span>
<span><span class="co">#&gt; 1         MYSFVSEETGTLIVNSVLLFLAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPSFYVYSRVKNLNSSRVPDLLV</span></span>
<span><span class="co">#&gt; 2        MYSFVSEETGTLIVNSVLLFLAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPTVYVYSRVKNLNSSEGVPDLLV</span></span>
<span><span class="co">#&gt; 3        MYSFVSEETGTLIVNSVLLFLAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPTVYVYSRVKNLNSSEGVPDLLV</span></span>
<span><span class="co">#&gt; 4        MYSFVSEETGTLIVNSVLLFLAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPTVYVYSRVKNLNSSEGVPDLLV</span></span>
<span><span class="co">#&gt; 5        MYSFVSEETGTLIVNSVLLFFAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPTVYVYSRVKNLNSSEGVPDLLV</span></span>
<span><span class="co">#&gt; 6  MLPFVHEQIGTIIVNFFILTVVCAITLLVCLAVLTAIRLCVQCASGVNTLLFVPAFYIYNTGRNAYFKFQENRPPFPPEDWV</span></span>
<span><span class="co">#&gt; 7  MLPFVHEQIGTIIVNFFILTVVCAITLVVCLAILTAIRLCVQCASGVNTLLFVPAFYIYNTGRNAYFKFQENRPPFPPEDWV</span></span>
<span><span class="co">#&gt; 8  MLPFVQERIGLFIVNFFIFTVVCAITLLVCMAFLTATRLCVQCMTGFNTLLVQPALYLYNTGRSVYVKFQDSKPPLPPDEWV</span></span>
<span><span class="co">#&gt; 9  MTFPRALTVIDDNGMVINIIFWFLLIIILILLSIALLNIIKLCMVCCNLGRTVIIVPAQHAYDAYKNFMRIKAYNPDGALLA</span></span>
<span><span class="co">#&gt; 10 MVDVFFTDTAWYVGQIFFLVLSCVIFLIFVVALLATIKLCIQICGFCNIFIISPSAYVYNRGRQLYKSYSEHVIPSTLDDLI</span></span>
<span><span class="co">#&gt; 11 MVDVFFTDTAWYVGQIFFLVLSCVIFLIFVVALLATIKLCIQICGFCNIFIISPSAYVYNRGRQLYKSYSEHVIPSTLDDLI</span></span>
<span><span class="co">#&gt; 12 MTFPRALTVIDDNGMVISIIFWFLLIIILILFSIALLNIIKLCMVCCNLGRTVIIVPARHAYDAYKNFMQIRAYNPDEALLV</span></span>
<span><span class="co">#&gt; 13 MLPFVQEQIGAFIVNFFILSVVCAVTLVVCLAILTAIRLCVQCVSGCHTLVFLPAVHIYNTGRAAYVKFQESHPPYPPEDWV</span></span>
<span><span class="co">#&gt; 14 MTFPRALTVIDDNGMVISIIFWFLLIIILILLSIALLNIIKLCMVCCNLGRTVIIVPVQHAYDAYKNFMRIKAYNPDGALLV</span></span>
<span><span class="co">#&gt; 15 MTFPRALTVIDDNGMVISIIFWFLLIIILILLSIALLNIIKLCMVCCNLGRTVIIVPVQHAYDAYKNFMRIKAYNPDGALLV</span></span></code></pre></div>
<p>By having the sequences as a column in our Data Frane, now we can preprocess it using Tidyverse “verbs”. This is how we could filter the sequences to keep only non-redundant sequences.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span></span>
<span>    <span class="va">hmmer.VEMP_SARS</span> <span class="op">&lt;-</span> <span class="va">hmmer.VEMP_SARS</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">hits.fullfasta</span>, .keep_all <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt;    algorithm                                 uuid stats.page stats.nhits</span></span>
<span><span class="co">#&gt; 1     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 2     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 3     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 4     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 5     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 6     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 7     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 8     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 9     phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 10    phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt; 11    phmmer 31D3D19C-6745-11ED-9EE3-3AE29E870EE0          1          43</span></span>
<span><span class="co">#&gt;    stats.elapsed stats.Z stats.Z_setby stats.n_past_msv stats.unpacked</span></span>
<span><span class="co">#&gt; 1           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 2           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 3           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 4           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 5           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 6           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 7           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 8           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 9           0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 10          0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt; 11          0.05  565928             0            24059             43</span></span>
<span><span class="co">#&gt;    stats.nseqs stats.user stats.domZ_setby stats.n_past_bias stats.sys</span></span>
<span><span class="co">#&gt; 1       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 2       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 3       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 4       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 5       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 6       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 7       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 8       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 9       565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 10      565928          0                0             14018         0</span></span>
<span><span class="co">#&gt; 11      565928          0                0             14018         0</span></span>
<span><span class="co">#&gt;    stats.n_past_fwd stats.total stats.nmodels stats.nincluded stats.n_past_vit</span></span>
<span><span class="co">#&gt; 1                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 2                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 3                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 4                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 5                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 6                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 7                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 8                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 9                60           1             1              35              902</span></span>
<span><span class="co">#&gt; 10               60           1             1              35              902</span></span>
<span><span class="co">#&gt; 11               60           1             1              35              902</span></span>
<span><span class="co">#&gt;    stats.nreported stats.domZ hits.archScore      hits.ph  hits.arch hits.kg</span></span>
<span><span class="co">#&gt; 1               43         43              1 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 2               43         43              1 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 3               43         43              1 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 4               43         43              2 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 5               43         43              2 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 6               43         43              2 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 7               43         43              0 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 8               43         43              0 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 9               43         43              0 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 10              43         43              2 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt; 11              43         43              0 Pisuviricota PF02723.17 Viruses</span></span>
<span><span class="co">#&gt;    hits.ndom                              hits.extlink hits.acc2 hits.taxid</span></span>
<span><span class="co">#&gt; 1          1 http://www.uniprot.org/uniprot/VEMP_SARS2    P0DTC4    2697049</span></span>
<span><span class="co">#&gt; 2          1 http://www.uniprot.org/uniprot/VEMP_BCHK3    Q3LZW9     442736</span></span>
<span><span class="co">#&gt; 3          1 http://www.uniprot.org/uniprot/VEMP_BC279    Q0Q473     389167</span></span>
<span><span class="co">#&gt; 4          1 http://www.uniprot.org/uniprot/VEMP_BC133    Q0Q4E8     389230</span></span>
<span><span class="co">#&gt; 5          1 http://www.uniprot.org/uniprot/VEMP_BCHK4    A3EX99     694007</span></span>
<span><span class="co">#&gt; 6          1 http://www.uniprot.org/uniprot/VEMP_MERS1    K9N5R3    1263720</span></span>
<span><span class="co">#&gt; 7          1 http://www.uniprot.org/uniprot/VEMP_CVPPU    P09048      11151</span></span>
<span><span class="co">#&gt; 8          1 http://www.uniprot.org/uniprot/VEMP_CVHN5    Q0ZME5     443241</span></span>
<span><span class="co">#&gt; 9          1 http://www.uniprot.org/uniprot/VEMP_CVCAI    P36696      36391</span></span>
<span><span class="co">#&gt; 10         1 http://www.uniprot.org/uniprot/VEMP_BCHK5    A3EXD5     694008</span></span>
<span><span class="co">#&gt; 11         1 http://www.uniprot.org/uniprot/VEMP_CVPR8    P69611      33736</span></span>
<span><span class="co">#&gt;      hits.acc                     hits.taxlink                       hits.desc</span></span>
<span><span class="co">#&gt; 1  VEMP_SARS2 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 2  VEMP_BCHK3 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 3  VEMP_BC279 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 4  VEMP_BC133 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 5  VEMP_BCHK4 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 6  VEMP_MERS1 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 7  VEMP_CVPPU http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 8  VEMP_CVHN5 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 9  VEMP_CVCAI http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 10 VEMP_BCHK5 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt; 11 VEMP_CVPR8 http://www.uniprot.org/taxonomy/ Envelope small membrane protein</span></span>
<span><span class="co">#&gt;    hits.pvalue hits.flags hits.nregions hits.niseqs  hits.name</span></span>
<span><span class="co">#&gt; 1   -108.52911          3             1          16 VEMP_SARS2</span></span>
<span><span class="co">#&gt; 2   -100.14785          3             1          16 VEMP_BCHK3</span></span>
<span><span class="co">#&gt; 3    -98.91970          3             1          16 VEMP_BC279</span></span>
<span><span class="co">#&gt; 4    -39.57741          3             1          16 VEMP_BC133</span></span>
<span><span class="co">#&gt; 5    -38.93211          3             1          16 VEMP_BCHK4</span></span>
<span><span class="co">#&gt; 6    -30.29100          3             1          16 VEMP_MERS1</span></span>
<span><span class="co">#&gt; 7    -29.06330          3             1          16 VEMP_CVPPU</span></span>
<span><span class="co">#&gt; 8    -28.90947          3             1          16 VEMP_CVHN5</span></span>
<span><span class="co">#&gt; 9    -27.84353          3             1          16 VEMP_CVCAI</span></span>
<span><span class="co">#&gt; 10   -27.36318          3             1          16 VEMP_BCHK5</span></span>
<span><span class="co">#&gt; 11   -26.76532          3             1          16 VEMP_CVPR8</span></span>
<span><span class="co">#&gt;                                                                                     hits.species</span></span>
<span><span class="co">#&gt; 1                                                Severe acute respiratory syndrome coronavirus 2</span></span>
<span><span class="co">#&gt; 2                                                                           Bat coronavirus HKU3</span></span>
<span><span class="co">#&gt; 3                                                                               Bat CoV 279/2005</span></span>
<span><span class="co">#&gt; 4                                                               Bat coronavirus (BtCoV/133/2005)</span></span>
<span><span class="co">#&gt; 5                                                                           Bat coronavirus HKU4</span></span>
<span><span class="co">#&gt; 6  Middle East respiratory syndrome-related coronavirus (isolate United Kingdom/H123990006/2012)</span></span>
<span><span class="co">#&gt; 7                              Porcine transmissible gastroenteritis coronavirus (strain Purdue)</span></span>
<span><span class="co">#&gt; 8                                                            Human coronavirus HKU1 (isolate N5)</span></span>
<span><span class="co">#&gt; 9                                                           Canine coronavirus (strain Insavc-1)</span></span>
<span><span class="co">#&gt; 10                                                                          Bat coronavirus HKU5</span></span>
<span><span class="co">#&gt; 11                          Porcine respiratory coronavirus (strain 86/137004 / isolate British)</span></span>
<span><span class="co">#&gt;    hits.score hits.bias hits.sindex hits.nincluded</span></span>
<span><span class="co">#&gt; 1       145.7       7.1     3311831              1</span></span>
<span><span class="co">#&gt; 2       134.2       6.4     3411115              1</span></span>
<span><span class="co">#&gt; 3       132.5       6.2     3410655              1</span></span>
<span><span class="co">#&gt; 4        51.2       3.8     3410660              1</span></span>
<span><span class="co">#&gt; 5        50.3       3.8     3411831              1</span></span>
<span><span class="co">#&gt; 6        38.4       2.9     3140808              1</span></span>
<span><span class="co">#&gt; 7        36.7       2.0     3128822              1</span></span>
<span><span class="co">#&gt; 8        36.5       1.9     3411155              1</span></span>
<span><span class="co">#&gt; 9        35.1       1.4     3138702              1</span></span>
<span><span class="co">#&gt; 10       34.4       4.0     3411906              1</span></span>
<span><span class="co">#&gt; 11       33.6       1.7     3138691              1</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   hits.domains</span></span>
<span><span class="co">#&gt; 1                                      1, VEMP_SARS2, 75, , 1, &gt;Query, 145.530029296875, 4.6e-42, 75, 1, 75, 7.10, 1, 3.5e-46, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnssrvpdllv, 1, 75, 1, , 3311831, 75, , , 0.99, MYSFVSEETGTLIVNSVLLFLAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPSFYVYSRVKNLNSSRVPDLLV, 75, 1, 8************************************************************************97, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnssrvpdllv, 75, 1, , 75, , Envelope small membrane protein, VEMP_SARS2, 1, 75</span></span>
<span><span class="co">#&gt; 2               1, VEMP_BCHK3, 72, , 1, &gt;Query, 134.040939331055, 2.0e-38, 76, 0.986666666666667, 76, 6.37, 1, 1.5e-42, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkp+ yvysrvknlnss+ vpdllv, 1, 76, 1, , 3411115, 76, , , 0.97, MYSFVSEETGTLIVNSVLLFLAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPTVYVYSRVKNLNSSEgVPDLLV, 75, 0.96, 8*****************************************************************98659***97, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnssr.vpdllv, 75, 1, , 74, , Envelope small membrane protein, VEMP_BCHK3, 1, 76</span></span>
<span><span class="co">#&gt; 3  1, VEMP_BC279, 71, , 1, &gt;Query, 132.357696533203, 6.8e-38, 76, 0.986666666666667, 76, 6.22, 1, 5.2e-42, mysfvseetgtlivnsvllf+afvvfllvtlailtalrlcayccnivnvslvkp+ yvysrvknlnss+ vpdllv, 1, 76, 1, , 3410655, 76, , , 0.97, MYSFVSEETGTLIVNSVLLFFAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPTVYVYSRVKNLNSSEgVPDLLV, 75, 0.946666666666667, 8*****************************************************************98659***97, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnssr.vpdllv, 75, 1, , 74, , Envelope small membrane protein, VEMP_BC279, 1, 76</span></span>
<span><span class="co">#&gt; 4                                                                     1, VEMP_BC133, 30, , 1, &gt;Query, 50.9613990783691, 4.2e-12, 64, 0.671875, 64, 3.81, 1, 3.2e-16, m  fv e+ gt+ivn  +l +   + llv la+lta+rlc  c++ vn  l  p+fy+y+  +n, 1, 82, 1, , 3410660, 71, , , 0.95, MLPFVHEQIGTIIVNFFILTVVCAITLLVCLAVLTAIRLCVQCASGVNTLLFVPAFYIYNTGRN, 64, 0.46875, 5679*******************************************************98877, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvkn, 75, 1, , 43, , Envelope small membrane protein, VEMP_BC133, 1, 64</span></span>
<span><span class="co">#&gt; 5                                                                     1, VEMP_BCHK4, 30, , 1, &gt;Query, 50.0769119262695, 8.0e-12, 64, 0.671875, 64, 3.78, 1, 6.1e-16, m  fv e+ gt+ivn  +l +   + l+v lailta+rlc  c++ vn  l  p+fy+y+  +n, 1, 82, 1, , 3411831, 71, , , 0.95, MLPFVHEQIGTIIVNFFILTVVCAITLVVCLAILTAIRLCVQCASGVNTLLFVPAFYIYNTGRN, 64, 0.46875, 5679*******************************************************98877, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvkn, 75, 1, , 43, , Envelope small membrane protein, VEMP_BCHK4, 1, 64</span></span>
<span><span class="co">#&gt; 6                                                      1, VEMP_MERS1, 25, , 1, &gt;Query, 38.2747611999512, 4.4e-08, 63, 0.603174603174603, 63, 2.90, 1, 3.4e-12, m  fv e  g +ivn  ++ +   + llv +a+lta rlc  c    n  lv+p++y+y+  +, 1, 82, 1, , 3140808, 73, , , 0.93, MLPFVQERIGLFIVNFFIFTVVCAITLLVCMAFLTATRLCVQCMTGFNTLLVQPALYLYNTGR, 63, 0.396825396825397, 5679************999999*************************************8755, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvk, 75, 1, , 38, , Envelope small membrane protein, VEMP_MERS1, 1, 63</span></span>
<span><span class="co">#&gt; 7                                                                        1, VEMP_CVPPU, 15, , 1, &gt;Query, 36.5278587341309, 1.6e-07, 69, 0.649122807017544, 69, 1.98, 6, 1.2e-11,    +++n ++ fl +++++l+++a+l  ++lc  ccn+    ++ p+ + y   kn+ , 9, 82, 1, , 3128822, 77, , , 0.90, N-GMVINIIFWFLLIIILILLSIALLNIIKLCMVCCNLGRTVIIVPAQHAYDAYKNFM, 66, 0.263157894736842, 4.4789***********************************************99974, tgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknln, 75, 13, , 37, , Envelope small membrane protein, VEMP_CVPPU, 13, 58</span></span>
<span><span class="co">#&gt; 8                                                                                       1, VEMP_CVHN5, 19, , 1, &gt;Query, 36.2906875610352, 1.9e-07, 68, 0.611111111111111, 68, 1.90, 3, 1.4e-11,    +l l+ v+fl+  +a+l  ++lc   c + n+ ++ ps yvy+r + l  s, 15, 82, 1, , 3411155, 75, , , 0.86, QIFFLVLSCVIFLIFVVALLATIKLCIQICGFCNIFIISPSAYVYNRGRQLYKS, 68, 0.351851851851852, 455788999**************************************9998766, nsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnss, 75, 15, , 33, , Envelope small membrane protein, VEMP_CVHN5, 15, 54</span></span>
<span><span class="co">#&gt; 9                                                            1, VEMP_CVCAI, 16, , 1, &gt;Query, 34.8620529174805, 5.3e-07, 72, 0.633333333333333, 72, 1.40, 6, 4.1e-11,  g ++++ ++ fl +++++l ++a+l  ++lc  ccn+    ++ p+ + y   kn+   r, 9, 82, 1, , 3138702, 79, , , 0.88, NG-MVISIIFWFLLIIILILFSIALLNIIKLCMVCCNLGRTVIIVPARHAYDAYKNFMQIR, 69, 0.266666666666667, 44.67889999********************************************987655, tgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknlnssr, 75, 13, , 38, , Envelope small membrane protein, VEMP_CVCAI, 13, 61</span></span>
<span><span class="co">#&gt; 10                                                               1, VEMP_BCHK5, 24, , 1, &gt;Query, 34.1980476379395, 8.7e-07, 61, 0.60655737704918, 61, 4.02, 1, 6.6e-11, m  fv e+ g +ivn  +l +   v l+v lailta+rlc  c +  +  +  p+ ++y+ , 1, 82, 1, , 3411906, 68, , , 0.94, MLPFVQEQIGAFIVNFFILSVVCAVTLVVCLAILTAIRLCVQCVSGCHTLVFLPAVHIYNT, 61, 0.39344262295082, 5679********************************************99999****9975, mysfvseetgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysr, 75, 1, , 37, , Envelope small membrane protein, VEMP_BCHK5, 1, 61</span></span>
<span><span class="co">#&gt; 11                                                                       1, VEMP_CVPR8, 15, , 1, &gt;Query, 33.3970336914062, 1.6e-06, 69, 0.649122807017544, 69, 1.73, 6, 1.2e-10,  g ++++ ++ fl +++++l+++a+l  ++lc  ccn+    ++ p  + y   kn+ , 9, 82, 1, , 3138691, 78, , , 0.88, NG-MVISIIFWFLLIIILILLSIALLNIIKLCMVCCNLGRTVIIVPVQHAYDAYKNFM, 66, 0.263157894736842, 44.67889999*****************************************999974, tgtlivnsvllflafvvfllvtlailtalrlcayccnivnvslvkpsfyvysrvknln, 75, 13, , 37, , Envelope small membrane protein, VEMP_CVPR8, 13, 58</span></span>
<span><span class="co">#&gt;                                                                         hits.pdbs</span></span>
<span><span class="co">#&gt; 1  7k3g_C, 7m4r_C, 7ntk_H, 7k3g_D, 7ntk_C, 7k3g_A, 7k3g_E, 7k3g_B, 7ntk_G, 7ntk_E</span></span>
<span><span class="co">#&gt; 2                                                                            NULL</span></span>
<span><span class="co">#&gt; 3                                                                            NULL</span></span>
<span><span class="co">#&gt; 4                                                                            NULL</span></span>
<span><span class="co">#&gt; 5                                                                            NULL</span></span>
<span><span class="co">#&gt; 6                                                                            NULL</span></span>
<span><span class="co">#&gt; 7                                                                            NULL</span></span>
<span><span class="co">#&gt; 8                                                                            NULL</span></span>
<span><span class="co">#&gt; 9                                                                            NULL</span></span>
<span><span class="co">#&gt; 10                                                                           NULL</span></span>
<span><span class="co">#&gt; 11                                                                           NULL</span></span>
<span><span class="co">#&gt;    hits.evalue hits.nreported         hits.archindex</span></span>
<span><span class="co">#&gt; 1        0e+00              1 88919621081425 2697049</span></span>
<span><span class="co">#&gt; 2        0e+00              1  88919621081425 442736</span></span>
<span><span class="co">#&gt; 3        0e+00              1  88919621081425 389167</span></span>
<span><span class="co">#&gt; 4        0e+00              1  88919621081425 389230</span></span>
<span><span class="co">#&gt; 5        7e-12              1  88919621081425 694007</span></span>
<span><span class="co">#&gt; 6        4e-08              1 88919621081425 1263720</span></span>
<span><span class="co">#&gt; 7        0e+00              1   88919621081425 11151</span></span>
<span><span class="co">#&gt; 8        0e+00              1  88919621081425 443241</span></span>
<span><span class="co">#&gt; 9        0e+00              1   88919621081425 36391</span></span>
<span><span class="co">#&gt; 10       1e-06              1  88919621081425 694008</span></span>
<span><span class="co">#&gt; 11       1e-06              1   88919621081425 33736</span></span>
<span><span class="co">#&gt;                                                                        hits.fullfasta</span></span>
<span><span class="co">#&gt; 1         MYSFVSEETGTLIVNSVLLFLAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPSFYVYSRVKNLNSSRVPDLLV</span></span>
<span><span class="co">#&gt; 2        MYSFVSEETGTLIVNSVLLFLAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPTVYVYSRVKNLNSSEGVPDLLV</span></span>
<span><span class="co">#&gt; 3        MYSFVSEETGTLIVNSVLLFFAFVVFLLVTLAILTALRLCAYCCNIVNVSLVKPTVYVYSRVKNLNSSEGVPDLLV</span></span>
<span><span class="co">#&gt; 4  MLPFVHEQIGTIIVNFFILTVVCAITLLVCLAVLTAIRLCVQCASGVNTLLFVPAFYIYNTGRNAYFKFQENRPPFPPEDWV</span></span>
<span><span class="co">#&gt; 5  MLPFVHEQIGTIIVNFFILTVVCAITLVVCLAILTAIRLCVQCASGVNTLLFVPAFYIYNTGRNAYFKFQENRPPFPPEDWV</span></span>
<span><span class="co">#&gt; 6  MLPFVQERIGLFIVNFFIFTVVCAITLLVCMAFLTATRLCVQCMTGFNTLLVQPALYLYNTGRSVYVKFQDSKPPLPPDEWV</span></span>
<span><span class="co">#&gt; 7  MTFPRALTVIDDNGMVINIIFWFLLIIILILLSIALLNIIKLCMVCCNLGRTVIIVPAQHAYDAYKNFMRIKAYNPDGALLA</span></span>
<span><span class="co">#&gt; 8  MVDVFFTDTAWYVGQIFFLVLSCVIFLIFVVALLATIKLCIQICGFCNIFIISPSAYVYNRGRQLYKSYSEHVIPSTLDDLI</span></span>
<span><span class="co">#&gt; 9  MTFPRALTVIDDNGMVISIIFWFLLIIILILFSIALLNIIKLCMVCCNLGRTVIIVPARHAYDAYKNFMQIRAYNPDEALLV</span></span>
<span><span class="co">#&gt; 10 MLPFVQEQIGAFIVNFFILSVVCAVTLVVCLAILTAIRLCVQCVSGCHTLVFLPAVHIYNTGRAAYVKFQESHPPYPPEDWV</span></span>
<span><span class="co">#&gt; 11 MTFPRALTVIDDNGMVISIIFWFLLIIILILLSIALLNIIKLCMVCCNLGRTVIIVPVQHAYDAYKNFMRIKAYNPDGALLV</span></span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="data-exploration">Data Exploration<a class="anchor" aria-label="anchor" href="#data-exploration"></a>
</h4>
<div class="section level5">
<h5 id="taxonomic-distribution">Taxonomic distribution<a class="anchor" aria-label="anchor" href="#taxonomic-distribution"></a>
</h5>
<p>Now we can explore the nature of these sequences. One aspect of great interest is the taxonomic distribution of these sequences. We can consult it because we have the <code>hits.taxid</code> parameter, which identifies taxonomic species. We proceed to taxonomically annotate the results using the <code>add_taxa_to_HMMER_tbl</code> function</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hmmer.VEMP_SARS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/add_taxa_to_hmmer_tbl.html">add_taxa_to_hmmer_tbl</a></span><span class="op">(</span><span class="va">hmmer.VEMP_SARS</span>, mode <span class="op">=</span> <span class="st">"local"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Registered S3 method overwritten by 'hoardr':</span></span>
<span><span class="co">#&gt;   method           from</span></span>
<span><span class="co">#&gt;   print.cache_info httr</span></span></code></pre></div>
<p>Note that we have indicated the parameter <code>mode = “local”</code> to note that we don’t want to use remote resources. Using remote resources can be slow, we advise you, if the amount of sequences is considerable, to use the “local” option to install (only once) and then use a local database.</p>
<p>Now, we can use all the meta library of tidyverse and ggplot2 to explore the data and make graphs. Let’s see an example of what could be done:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">hmmer.VEMP_SARS</span>,</span>
<span>    mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">hits.evalue</span>, fill <span class="op">=</span> <span class="va">taxa.subgenus</span><span class="op">)</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span>binwidth <span class="op">=</span> <span class="fl">10</span><span class="op">^</span><span class="op">-</span><span class="fl">8</span><span class="op">)</span></span></code></pre></div>
<p><img src="HMMERutils_files/figure-html/taxa_plot-1.png" width="1458.33333333333"></p>
</div>
<div class="section level5">
<h5 id="calculating-the-identity-of-pairwise-alignments">Calculating the identity of pairwise alignments<a class="anchor" aria-label="anchor" href="#calculating-the-identity-of-pairwise-alignments"></a>
</h5>
<p>To examine the diversity of the sequences we can align all the sequences in pairs and then calculate the percentages of identity for each one using the <code>pairwise_alignment_sequence_identity</code> function.</p>
<p>When calling this function, we need to specify which sequences we want to compare (as a vector of characters), the type of alignment, and the type of sequence identity we are going to use. In this case, we will use the default options: <code>aln_type = "global"</code> (align whole strings with end gap penalties by using the Needleman-Wunsch algorithm) and <code>pid_type = "PID1"</code> ( PID1=100⋅identical position aligned positions + internal gap positions).</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">percentage_identities</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pairwise_alignment_sequence_identity.html">pairwise_alignment_sequence_identity</a></span><span class="op">(</span></span>
<span>    seqs <span class="op">=</span> <span class="va">hmmer.VEMP_SARS</span><span class="op">$</span><span class="va">hits.fullfasta</span>,</span>
<span>    aln_type <span class="op">=</span> <span class="st">"global"</span>,</span>
<span>    pid_type <span class="op">=</span> <span class="st">"PID1"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Now, we can inspect the distribution of those values in a histogram:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/pairwise_sequence_identity_histogram.html">pairwise_sequence_identity_histogram</a></span><span class="op">(</span><span class="va">percentage_identities</span><span class="op">)</span></span></code></pre></div>
<p><img src="HMMERutils_files/figure-html/hist-1.png" width="1458.33333333333"></p>
<p>And, to compare this information with other parameters, a heatmap is. Note that we use the <code>taxa.subgenus</code> column to annotate the heatmap, for which we use the <code>annotation</code> argument and the <code>$</code> operator.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/pairwise_sequence_identity_heatmap.html">pairwise_sequence_identity_heatmap</a></span><span class="op">(</span></span>
<span>    <span class="va">percentage_identities</span>,</span>
<span>    annotation <span class="op">=</span> <span class="va">hmmer.VEMP_SARS</span><span class="op">$</span><span class="va">taxa.subgenus</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="HMMERutils_files/figure-html/unnamed-chunk-2-1.png" width="1458.33333333333"></p>
</div>
<div class="section level5">
<h5 id="physicochemical-properties">Physicochemical properties<a class="anchor" aria-label="anchor" href="#physicochemical-properties"></a>
</h5>
<p>We can study what the sequences we have obtained are like in more detail by calculating physicochemical properties based on their primary sequence. To do this, we can use the <code>add_physicochemical_properties_to_HMMER_tbl</code> function</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hmmer.VEMP_SARS</span> <span class="op">&lt;-</span> <span class="va">hmmer.VEMP_SARS</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/add_physicochemical_properties_to_HMMER_tbl.html">add_physicochemical_properties_to_HMMER_tbl</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Let’s see how are these new properties distributed along our sequences:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hmmer.VEMP_SARS</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">taxa.subgenus</span>, y <span class="op">=</span> <span class="va">properties.molecular.weight</span>, fill <span class="op">=</span> <span class="va">taxa.subgenus</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="HMMERutils_files/figure-html/boxplot-1.png" width="1458.33333333333"></p>
</div>
</div>
<div class="section level4">
<h4 id="proteins-homologous-to-ns8a_sarsin-swissprot-and-pdb-using-phmmer">Proteins homologous to NS8A_SARSin SwissProt and PDB using PHMMER<a class="anchor" aria-label="anchor" href="#proteins-homologous-to-ns8a_sarsin-swissprot-and-pdb-using-phmmer"></a>
</h4>
<p>So far, we have been working with the results of a single search. This case study has allowed us to describe in depth most of the utilities of this library. However, the most common scenario is that we want to work with the results of several searches. Let’s try performing a search with PHMMER for the sequence of NS8A_SARS against SwissProt and PDB.</p>
<p>For this example, moreover, we will not go step by step but will show how HMMERutils can perform all of the above operations in a few lines of code. You can refer to the information regarding the <code>search_phmmer</code> function using <code><a href="../reference/search_phmmer.html">?search_phmmer</a></code> for a complete listing of options.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span></span>
<span>    <span class="va">hmmer.NS8A_SARS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/search_phmmer.html">search_phmmer</a></span><span class="op">(</span></span>
<span>        seq <span class="op">=</span> <span class="va">NS8A_SARS</span>,</span>
<span>        seqdb <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pdb"</span>, <span class="st">"swissprot"</span><span class="op">)</span>,</span>
<span>        verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 9 × 48</span></span></span>
<span><span class="co">#&gt;   algori…¹ uuid  stats…² stats…³ stats…⁴ stats.Z stats…⁵ stats…⁶ stats…⁷ stats…⁸</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> phmmer   3BE5…       1       2 0.08     <span style="text-decoration: underline;">546</span>676       0    <span style="text-decoration: underline;">2</span>698       2  <span style="text-decoration: underline;">546</span>676</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> phmmer   3BE5…       1       2 0.08     <span style="text-decoration: underline;">546</span>676       0    <span style="text-decoration: underline;">2</span>698       2  <span style="text-decoration: underline;">546</span>676</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> phmmer   3C56…       1       7 0.09     <span style="text-decoration: underline;">565</span>928       0   <span style="text-decoration: underline;">10</span>855       7  <span style="text-decoration: underline;">565</span>928</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> phmmer   3C56…       1       7 0.09     <span style="text-decoration: underline;">565</span>928       0   <span style="text-decoration: underline;">10</span>855       7  <span style="text-decoration: underline;">565</span>928</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> phmmer   3C56…       1       7 0.09     <span style="text-decoration: underline;">565</span>928       0   <span style="text-decoration: underline;">10</span>855       7  <span style="text-decoration: underline;">565</span>928</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> phmmer   3C56…       1       7 0.09     <span style="text-decoration: underline;">565</span>928       0   <span style="text-decoration: underline;">10</span>855       7  <span style="text-decoration: underline;">565</span>928</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> phmmer   3C56…       1       7 0.09     <span style="text-decoration: underline;">565</span>928       0   <span style="text-decoration: underline;">10</span>855       7  <span style="text-decoration: underline;">565</span>928</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">8</span> phmmer   3C56…       1       7 0.09     <span style="text-decoration: underline;">565</span>928       0   <span style="text-decoration: underline;">10</span>855       7  <span style="text-decoration: underline;">565</span>928</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">9</span> phmmer   3C56…       1       7 0.09     <span style="text-decoration: underline;">565</span>928       0   <span style="text-decoration: underline;">10</span>855       7  <span style="text-decoration: underline;">565</span>928</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 38 more variables: stats.user &lt;dbl&gt;, stats.domZ_setby &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   stats.n_past_bias &lt;int&gt;, stats.sys &lt;dbl&gt;, stats.n_past_fwd &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   stats.total &lt;dbl&gt;, stats.nmodels &lt;int&gt;, stats.nincluded &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   stats.n_past_vit &lt;int&gt;, stats.nreported &lt;int&gt;, stats.domZ &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   hits.archScore &lt;chr&gt;, hits.ph &lt;chr&gt;, hits.arch &lt;chr&gt;, hits.kg &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   hits.ndom &lt;int&gt;, hits.extlink &lt;chr&gt;, hits.taxid &lt;chr&gt;, hits.acc &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   hits.taxlink &lt;chr&gt;, hits.desc &lt;chr&gt;, hits.pvalue &lt;dbl&gt;, hits.seqs &lt;list&gt;, …</span></span></span></code></pre></div>
<p>Note that, now, the number of searches performed is 2:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">hmmer.NS8A_SARS</span><span class="op">$</span><span class="va">uuid</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "3BE570D2-6745-11ED-A01F-0B0CB769042E"</span></span>
<span><span class="co">#&gt; [2] "3C5697B2-6745-11ED-A01F-0B0CB769042E"</span></span></code></pre></div>
<p>Now, using the pipe operator, we are going to filter the sequences, read the sequences, annotate them taxonomically, and calculate the physicochemical properties of the sequences.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hmmer.NS8A_SARS</span> <span class="op">&lt;-</span> <span class="va">hmmer.NS8A_SARS</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/filter_hmmer.html">filter_hmmer</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/add_sequences_to_hmmer_tbl.html">add_sequences_to_hmmer_tbl</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/add_taxa_to_hmmer_tbl.html">add_taxa_to_hmmer_tbl</a></span><span class="op">(</span>mode <span class="op">=</span> <span class="st">"local"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/add_physicochemical_properties_to_HMMER_tbl.html">add_physicochemical_properties_to_HMMER_tbl</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="section level5">
<h5 id="extracting-nested-hashes">Extracting nested hashes<a class="anchor" aria-label="anchor" href="#extracting-nested-hashes"></a>
</h5>
<p>As we have searched against PDB, now we have a new column, which is called <code>hits.pdbs</code>. To explore the PDB files available for homologous sequences we can “extract” those nested rows.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pdbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_from_hmmer.html">extract_from_hmmer</a></span><span class="op">(</span><span class="va">hmmer.NS8A_SARS</span>, column <span class="op">=</span> <span class="st">"hits.pdbs"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">pdbs</span><span class="op">$</span><span class="va">pdbs</span><span class="op">)</span></span>
<span><span class="co">#&gt; [[1]]</span></span>
<span><span class="co">#&gt; [1] "7jx6_A"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span>
<span><span class="co">#&gt; [1] NA</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[3]]</span></span>
<span><span class="co">#&gt; [1] "7jtl_B"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[4]]</span></span>
<span><span class="co">#&gt; [1] "7jtl_A"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[5]]</span></span>
<span><span class="co">#&gt; [1] "7jx6_B"</span></span></code></pre></div>
<p>We can do the same to study the domains (default behavior):</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">domains</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_from_hmmer.html">extract_from_hmmer</a></span><span class="op">(</span><span class="va">hmmer.NS8A_SARS</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">domains</span><span class="op">$</span><span class="va">domains.alisqdesc</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "ORF8 protein"             "Non-structural protein 8"</span></span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="using-hmmsearch">Using hmmsearch<a class="anchor" aria-label="anchor" href="#using-hmmsearch"></a>
</h4>
<p>So far, we have only used PHMMER, since it is the most used and default algorithm. However, HMMERutils allows us to work with all HMMER algorithms. Let’s see how it can be used to perform a search using hmmsearch. This algorithm performs a sequence search in the same databases as PHMMER with the difference that, instead of building a hidden Markov model with a single sequence, it does it with an alignment.</p>
<p>For this example, we will use a fake alignment (that’s why we are creating it from a vector, instead of reading an actual file, using the <code>readAAMultipleAlignment</code> function.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alignment</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html" class="external-link">AAMultipleAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"FQTWEEFSRAAEKLYLADPMKVRVVLKYRHVDGNLCIKVTDDLVCLVYRTDQAQDVKKIEKF"</span>,</span>
<span>    <span class="st">"KYRTWEEFTRAAEKLYQADPMKVRVVLKYRHCDGNLCIKVTDDVVCLLYRTDQAQDVKKIEK"</span>,</span>
<span>    <span class="st">"EEYQTWEEFARAAEKLYLTDPMKVRVVLKYRHCDGNLCMKVTDDAVCLQYKTDQAQDVKKVE"</span>,</span>
<span>    <span class="st">"EEFSRAVEKLYLTDPMKVRVVLKYRHCDGNLCIKVTDNSVVSYEMRLFGVQKDNFALEHSLL"</span>,</span>
<span>    <span class="st">"SWEEFAKAAEVLYLEDPMKCRMCTKYRHVDHKLVVKLTDNHTVLKYVTDMAQDVKKIEKLTT"</span>,</span>
<span>    <span class="st">"FTNWEEFAKAAERLHSANPEKCRFVTKYNHTKGELVLKLTDDVVCLQYSTNQLQDVKKLEKL"</span>,</span>
<span>    <span class="st">"SWEEFVERSVQLFRGDPNATRYVMKYRHCEGKLVLKVTDDRECLKFKTDQAQDAKKMEKLNN"</span></span>
<span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Now, let’s perform a query using hmmsearch:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fake_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/search_hmmsearch.html">search_hmmsearch</a></span><span class="op">(</span><span class="va">alignment</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="using-hmmscan">Using HMMSCAN<a class="anchor" aria-label="anchor" href="#using-hmmscan"></a>
</h4>
<p>Another HMMER algorithm that can be used through HMMERutils is HMMSCAN. This algorithm performs a search of a protein sequence against an HMM database and can be used to identify domains in our sequence. Let’s see which domains we can find in VEMP_SARS:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span></span>
<span>    <span class="va">scan.VEMP_SARS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/search_hmmscan.html">search_hmmscan</a></span><span class="op">(</span><span class="va">VEMP_SARS</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 36</span></span></span>
<span><span class="co">#&gt;   algori…¹ uuid  stats…² stats…³ stats…⁴ stats.Z stats…⁵ stats…⁶ stats…⁷ stats…⁸</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> hmmscan  3F02…       1       1 0.06      <span style="text-decoration: underline;">19</span>632       0    <span style="text-decoration: underline;">1</span>297       1       1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 26 more variables: stats.user &lt;dbl&gt;, stats.domZ_setby &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   stats.n_past_bias &lt;int&gt;, stats.sys &lt;dbl&gt;, stats.n_past_fwd &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   stats.total &lt;dbl&gt;, stats.nmodels &lt;int&gt;, stats.nincluded &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   stats.n_past_vit &lt;int&gt;, stats.nreported &lt;int&gt;, stats.domZ &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   hits.flags &lt;int&gt;, hits.nregions &lt;int&gt;, hits.ndom &lt;int&gt;, hits.name &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   hits.score &lt;chr&gt;, hits.bias &lt;chr&gt;, hits.taxid &lt;chr&gt;, hits.acc &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   hits.domains &lt;list&gt;, hits.nincluded &lt;int&gt;, hits.evalue &lt;dbl&gt;, …</span></span></span>
<span><span class="va">scan.VEMP_SARS</span><span class="op">$</span><span class="va">hits.acc</span></span>
<span><span class="co">#&gt; [1] "PF02723.17"</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.16/HMMERutils" class="external-link">HMMERutils</a></em> package was made possible thanks to:</p>
<ul>
<li>R (R Core Team, 2022)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.16/BiocStyle" class="external-link">BiocStyle</a></em> (Oleś, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em> (McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em> (Allaire, Xie, McPherson, Luraschi, Ushey, Atkins, Wickham, Cheng, Chang, and Iannone, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em> (Wickham, Chang, Flight, Müller, and Hester, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em> (Wickham, 2011)</li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.16/biocthis" class="external-link">biocthis</a></em>.</p>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create the vignette</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rstudio/rmarkdown" class="external-link">"rmarkdown"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html" class="external-link">render</a></span><span class="op">(</span><span class="st">"HMMERutils.Rmd"</span>, <span class="st">"BiocStyle::html_document"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Extract the R code</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/knitr/" class="external-link">"knitr"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html" class="external-link">knit</a></span><span class="op">(</span><span class="st">"HMMERutils.Rmd"</span>, tangle <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Date the vignette was generated.</p>
<pre><code><span><span class="co">#&gt; [1] "2022-11-18 14:31:02 CET"</span></span></code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code><span><span class="co">#&gt; Time difference of 28.821 secs</span></span></code></pre>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.2.2 (2022-10-31)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 20.04.5 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  C.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    C.UTF-8</span></span>
<span><span class="co">#&gt;  tz       Europe/Copenhagen</span></span>
<span><span class="co">#&gt;  date     2022-11-18</span></span>
<span><span class="co">#&gt;  pandoc   2.5 @ /usr/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">!</span> <span style="color: #555555; font-style: italic;">package         </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;    ape                5.6-2    <span style="color: #555555;">2022-03-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> assertthat         0.2.1    <span style="color: #555555;">2019-03-21</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> backports          1.4.1    <span style="color: #555555;">2021-12-13</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> bibtex             0.5.0    <span style="color: #555555;">2022-09-25</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> BiocGenerics     * 0.42.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> BiocManager        1.30.19  <span style="color: #555555;">2022-10-25</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> BiocStyle        * 2.24.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> Biostrings       * 2.64.1   <span style="color: #555555;">2022-08-18</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> bit                4.0.4    <span style="color: #555555;">2020-08-04</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> bit64              4.0.5    <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> bitops             1.0-7    <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> blob               1.2.3    <span style="color: #555555;">2022-04-10</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    bold               1.2.0    <span style="color: #555555;">2021-05-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> bookdown           0.29     <span style="color: #555555;">2022-09-12</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> bslib              0.4.0    <span style="color: #555555;">2022-07-16</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> cachem             1.0.6    <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> cli                3.4.1    <span style="color: #555555;">2022-09-23</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> codetools          0.2-18   <span style="color: #555555;">2020-11-04</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    colorspace         2.0-3    <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    conditionz         0.1.0    <span style="color: #555555;">2019-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> crayon             1.5.2    <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    crul               1.3      <span style="color: #555555;">2022-09-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> curl               4.3.3    <span style="color: #555555;">2022-10-06</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    data.table         1.14.4   <span style="color: #555555;">2022-10-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> DBI                1.1.3    <span style="color: #555555;">2022-06-18</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> dbplyr             2.2.1    <span style="color: #555555;">2022-06-27</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> desc               1.4.2    <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> digest             0.6.30   <span style="color: #555555;">2022-10-18</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> dplyr            * 1.0.10   <span style="color: #555555;">2022-09-01</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> ellipsis           0.3.2    <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> evaluate           0.17     <span style="color: #555555;">2022-10-07</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> fansi              1.0.3    <span style="color: #555555;">2022-03-24</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    farver             2.1.1    <span style="color: #555555;">2022-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> fastmap            1.1.0    <span style="color: #555555;">2021-01-25</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    foreach            1.5.2    <span style="color: #555555;">2022-02-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> fs                 1.5.2    <span style="color: #555555;">2021-12-08</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> generics           0.1.3    <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> GenomeInfoDb     * 1.32.4   <span style="color: #555555;">2022-09-06</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> GenomeInfoDbData   1.2.8    <span style="color: #555555;">2022-11-06</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    ggplot2          * 3.4.0    <span style="color: #555555;">2022-11-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> glue               1.6.2    <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    gtable             0.3.1    <span style="color: #555555;">2022-09-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> highr              0.9      <span style="color: #555555;">2021-04-16</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    HMMERutils       * 0.99.0   <span style="color: #555555;">2022-11-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> hoardr             0.5.2    <span style="color: #555555;">2018-12-02</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> htmltools          0.5.3    <span style="color: #555555;">2022-07-18</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    httpcode           0.3.0    <span style="color: #555555;">2020-04-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> httr               1.4.4    <span style="color: #555555;">2022-08-17</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> IRanges          * 2.30.1   <span style="color: #555555;">2022-08-18</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    iterators          1.0.14   <span style="color: #555555;">2022-02-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> jquerylib          0.1.4    <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> jsonlite           1.8.3    <span style="color: #555555;">2022-10-21</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> knitr              1.40     <span style="color: #555555;">2022-08-24</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    labeling           0.4.2    <span style="color: #555555;">2020-10-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> lattice            0.20-45  <span style="color: #555555;">2021-09-22</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> lifecycle          1.0.3    <span style="color: #555555;">2022-10-07</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> lubridate          1.8.0    <span style="color: #555555;">2021-10-07</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> magrittr           2.0.3    <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> memoise            2.0.1    <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    munsell            0.5.0    <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> nlme               3.1-160  <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> Peptides           2.4.4    <span style="color: #555555;">2021-05-14</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    pheatmap           1.0.12   <span style="color: #555555;">2019-01-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> pillar             1.8.1    <span style="color: #555555;">2022-08-19</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> pkgconfig          2.0.3    <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> pkgdown            2.0.6    <span style="color: #555555;">2022-07-16</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> plyr               1.8.7    <span style="color: #555555;">2022-03-24</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> purrr              0.3.5    <span style="color: #555555;">2022-10-06</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> R6                 2.5.1    <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> ragg               1.2.4    <span style="color: #555555;">2022-10-24</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> rappdirs           0.3.3    <span style="color: #555555;">2021-01-31</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    RColorBrewer       1.1-3    <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> Rcpp               1.0.9    <span style="color: #555555;">2022-07-08</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> RCurl              1.98-1.9 <span style="color: #555555;">2022-10-03</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> RefManageR       * 1.4.0    <span style="color: #555555;">2022-09-30</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    renv               0.16.0   <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    reshape            0.8.9    <span style="color: #555555;">2022-04-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> rlang              1.0.6    <span style="color: #555555;">2022-09-24</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> rmarkdown          2.17     <span style="color: #555555;">2022-10-07</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> rprojroot          2.0.3    <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> RSQLite            2.2.18   <span style="color: #555555;">2022-10-04</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> S4Vectors        * 0.34.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> sass               0.4.2    <span style="color: #555555;">2022-07-16</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    scales             1.2.1    <span style="color: #555555;">2022-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> sessioninfo      * 1.2.2    <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> stringi            1.7.8    <span style="color: #555555;">2022-07-11</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> stringr            1.4.1    <span style="color: #555555;">2022-08-20</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> systemfonts        1.0.4    <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    taxize             0.9.100  <span style="color: #555555;">2022-04-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> taxizedb           0.3.0    <span style="color: #555555;">2021-01-15</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> textshaping        0.3.6    <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> tibble             3.1.8    <span style="color: #555555;">2022-07-22</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> tidyr              1.2.1    <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> tidyselect         1.2.0    <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> utf8               1.2.2    <span style="color: #555555;">2021-07-24</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    uuid               1.1-0    <span style="color: #555555;">2022-04-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;    vctrs              0.5.0    <span style="color: #555555;">2022-10-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> withr              2.5.0    <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> xfun               0.34     <span style="color: #555555;">2022-10-18</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> xml2               1.3.3    <span style="color: #555555;">2021-11-30</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> XVector          * 0.36.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> yaml               2.3.6    <span style="color: #555555;">2022-10-18</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> zlibbioc           1.42.0   <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[?]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;    zoo                1.8-11   <span style="color: #555555;">2022-09-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.2)</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /home/currocam/.cache/R/renv/library/HMMERutils-c54e77b1/R-4.2/x86_64-pc-linux-gnu</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /home/currocam/Code/HMMERutils/renv/sandbox/R-4.2/x86_64-pc-linux-gnu/9a444a72</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  <span style="color: #BBBBBB; background-color: #BB0000;">P</span> ── Loaded and on-disk path mismatch.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.16/BiocStyle" class="external-link">BiocStyle</a></em> (Oleś, 2022) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie, 2022) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em> (Allaire, Xie, McPherson et al., 2022) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em> (McLean, 2017).</p>
<p>
<a id="bib-allaire2022rmarkdown"></a><a href="#cite-allaire2022rmarkdown">[1]</a><cite> J. Allaire, Y. Xie, J. McPherson, et al. <em>rmarkdown: Dynamic Documents for R</em>. R package version 2.17. 2022. URL: <a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[2]</a><cite> M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX References in R”. In: <em>The Journal of Open Source Software</em> (2017). DOI: <a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-ole2022biocstyle"></a><a href="#cite-ole2022biocstyle">[3]</a><cite> A. Oleś. <em>BiocStyle: Standard styles for vignettes and other Bioconductor documents</em>. R package version 2.24.0. 2022. URL: <a href="https://github.com/Bioconductor/BiocStyle" class="external-link">https://github.com/Bioconductor/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2022language"></a><a href="#cite-2022language">[4]</a><cite> R Core Team. <em>R: A Language and Environment for Statistical Computing</em>. R Foundation for Statistical Computing. Vienna, Austria, 2022. URL: <a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[5]</a><cite> H. Wickham. “testthat: Get Started with Testing”. In: <em>The R Journal</em> 3 (2011), pp. 5–10. URL: <a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2021sessioninfo"></a><a href="#cite-wickham2021sessioninfo">[6]</a><cite> H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session Information</em>. R package version 1.2.2. 2021. URL: <a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">https://CRAN.R-project.org/package=sessioninfo</a>.</cite>
</p>
<p>
<a id="bib-xie2022knitr"></a><a href="#cite-xie2022knitr">[7]</a><cite> Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report Generation in R</em>. R package version 1.40. 2022. URL: <a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Francisco de Borja Campuzano Jiménez.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
